<!DOCTYPE html><html><head>
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
<meta content="text/html; charset=utf-8" http-equiv="Content-Type">
<meta name="description" content="iCn3D Structure Viewer">
<meta name="ncbi_db" content="structure">
<meta name="keywords" content="NCBI, Structure, JavaScript, iCn3D, 3D, Viewer, WebGL, three.js, sequence, chemical">
<meta name="robots" content="index,follow,noarchive">
<meta name="ncbi_app" content="icn3d">
<meta name="ncbi_pdid" content="webapi">
<meta name="ncbi_db" content="structure">
<meta name="ncbi_pinger_xml_http_request_override" content="false"/>
<title>iCn3D: Web-based 3D Structure Viewer</title>
<script type="text/javascript">
    window.ncbi_startTime = new Date();
</script>

<link rel="stylesheet" href="lib/jquery-ui-1.12.1.min.css">
<link rel="stylesheet" href="icn3d_full_ui.css">
<script src="lib/jquery-3.5.0.min.js"></script>
<script src="lib/jquery-ui-1.12.1.min.js"></script>
<script src="lib/three_0.103.0.min.js"></script>
<script src="icn3d_full_ui.min.js"></script>

<style type="text/css">
pre { white-space: pre-wrap; }
A { text-decoration: none;}

A:link { text-decoration: none; font-family: Verdana, Arial, Sans-serif; font-size: 100%; color:#0056CC; }
A:visited { text-decoration: none; font-family: Verdana, Arial, Sans-serif; font-size: 100%;  color:#00994E; }
A:active { text-decoration: none; font-family: Verdana, Arial, Sans-serif; font-size: 100%;  color:#00994E; }
A:hover { text-decoration: none; color:black;  font-family: Verdana, Arial, Sans-serif; font-size: 100%; color:#00994E;}

.top {
  width: 12px;
  height: 12px;
  background: url(https://www.ncbi.nlm.nih.gov/Structure/IMG/arrowup_blue.gif) no-repeat;
  margin-top: 6px;
}

.indent {margin-left:40px;}

.gallery {float:left; border: solid 1px #0000ff; padding: 5px; margin: 10px; text-align:center;}

body {font-family: Verdana, Arial, Helvetica, sans-serif;}
</style>

</head>
<body>
<a name="Top"></a>
<blockquote>
<h1>iCn3D Web APIs</h1>

<!-- ============= TABLE_OF_CONTENTS ============= -->

<ul>

<li><b><a href="#gallery">iCn3D gallery with live examples</a></b></li>
<br>

<!--li>What is iCn3D Structure Viewer?</li><br/>
"I see in 3D" (iCn3D) Structure Viewer is a WebGL-based 3D viewer using Three.js and jQuery. iCn3D has a feature-rich user interface and allows users to:<br/><br/>
<ul>
<li><a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/full.html?mmdbid=1tup&command=view annotations; set annotation cdd; set view detailed view; set annotation all">Display and integrate annotations</a> from NCBI resources including dbSNP, ClinVar, conserved domains, 3D domains, and binding sites</li>
<li><a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/full.html?mmdbid=1tup&showanno=1&show2d=1&showsets=1&command=view annotations; set view detailed view; add track | chainid 1TUP_B | title Custom Key Sites | text 82 R, 152 G, 155-156 RR, 180 R, 189 R; select .B:82,152,155-156,180,189 | name mutation">Add custom tracks</a> in various formats (FASTA, bed file, etc) in the annotation window</li>
<li>Use a url (e.g., <a href="https://structure.ncbi.nlm.nih.gov/icn3d/share.html?uRL4mitQdG6J3gGg8">https://structure.ncbi.nlm.nih.gov/icn3d/share.html?uRL4mitQdG6J3gGg8</a>) or a state file to capture the custom display of 3D structures. The saved static "iCn3D PNG Image" can be loaded using "File > Open File > iCn3D PNG Image" to reproduce the interactive display.</li>
<li>Select residues by searching sequences or select on 3D structures, 2D interactions, and 1D sequences</li>
<li>Display/highlight selected residues in 3D structures, 2D interactions, and 1D sequences</li>
<li>Show <a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/full.html?mmdbid=3gvu&command=style+proteins+b+factor+tube;+color+b+factor;+set+map+2fofc+sigma+1.5">electron density map</a> or <a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/full.html?mmdbid=6eny&command=set+emmap+percentage+20">EM density map</a></li>
<li>Export STL or VRML files for <a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/full.html?mmdbid=1tup&command=export stl stabilizer file">3D printing</a></li>
<li>Display NCBI pre-calculated <a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/full.html?align=1hho,4n7n&showalignseq=1">aligned 3D structures</a> by providing two PDB IDs or MMDB IDs</li>
</ul>
<br/>

We provided two types of iCn3D widgets: <a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/index.html?mmdbid=1tup">basic interface</a> and <a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/full.html?mmdbid=1tup">advanced interface</a>.<br><br/>
<ol>
<li>The basic interface has the minimum javascript code for the interface. It has the basic features such as changing color and styles.</li>
<li>The advanced interface has a library for the interface. It is recommended since it has many features and can have simple UI using <a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d.html#parameters" target="_blank">URL parameters</a> such as <a href="https://structure.ncbi.nlm.nih.gov/icn3d/share.html?2J9Zv4i74yxyagtFA" target="_blank">https://structure.ncbi.nlm.nih.gov/icn3d/share.html?2J9Zv4i74yxyagtFA</a>.</li>
</ol>
<br/>

This page describes Web APIs and how to embed iCn3D in your html page. Other details are described in the <a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/docs/icn3d_help.html">help</a> page. The source code of iCn3D is available at <a href="https://github.com/ncbi/icn3d">github.com/ncbi/icn3d</a>.<br/>
<br/-->

<li><b>What is iCn3D Structure Viewer?</b></li><br/>
"I see in 3D" (iCn3D) Structure Viewer is a WebGL-based 3D viewer using Three.js and jQuery. iCn3D synchronizes the display of 3D structure, 2D interaction, and 1D sequences and annotations. Users' custom display can be saved in a short URL or a PNG image.<br/><br/>
<ol>
<li><b>View a 3D structure in iCn3D</b><br/>
Open the link https://www.ncbi.nlm.nih.gov/Structure/icn3d/full.html, input a PDB ID, and click "Load". You can also click "File" menu to "Open File" or input other IDs.<br/>
<br/>
As mentioned in the menu "Help > Transformation Hints", you can use Left mouse button for rotation, Middle mouse wheel for zooming, and Right mouse button for translation. <br>
<br>
The most important point about using iCn3D is the current selection. Any operations on color, style, etc. are working on the current selection. By default, all atoms are selected. Once you select any subset, your operation will work ONLY on the subset. You can switch the selection using the toggle next to the Help menu.<br/>
<br/>
<li><b>Create custom 3D view</b><br/>
You first open a structure in "File" menu, then select a subset in "Select" menu, view only the selected subset by clicking "View Only Selection" in View menu, finally change styles or colors in "Style" and "Color" menus. <br/>
<br/>
Each operation has a corresponding command as listed at https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d.html#commands. These commands will show up in the command/log window right beneath the 3D display. To view all previous commands, you can click "Share Link" in "File" menu. Both the original URL and the short URL can be used to display your custom view.<br/>
<br/>
<li><b>Save your work</b><br/>
You can save "iCn3D PNG Image" in the menu "File > Save Files". Both the PNG file and an HTML file are saved. Click the HTML file to see the PNG image, which is linked to the custom display via a shorten URL. The downloaded "iCn3D PNG Image" itself can also be used as an input in the menu "File > Open File" to reproduce the custom display. You can combine these HTML files to generate your own galleries.<br/>
<br/>
You can also save "Share Link" in "File" menu to share with your colleagues. You can click "Replay Each Step > On" in "File" menu to learn how a <a href="https://structure.ncbi.nlm.nih.gov/icn3d/share.html?u7gp4xS9rn4hahcLA">custom display</a> was generated.<br/>
<br/>
<li><b><a href="https://structure.ncbi.nlm.nih.gov/icn3d/share.html?JR5B">Show binding site</a></b><br/>
You can click "Chem. Binding" in "View" menu to show all hydrogen bonds around chemicals. You can also <a href="https://structure.ncbi.nlm.nih.gov/icn3d/share.html?CuXYgGLCukDeUKnJ6">Show interaction interface</a> by clicking "H-Bonds & Interactions > 2D Interaction Graph, or Highlight Interactionsin Table" in "View" menu.<br/>
<br/>
<li><b><a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/full.html?mmdbid=1tup&command=export+stl+stabilizer+file">Export models for 3D printing</a></b><br/>
You can click "3D Printing" in "File" menu to export models for 3D printing. Both STL and VRML files are supported.<br/>
<br/>
<li><b><a href="https://structure.ncbi.nlm.nih.gov/icn3d/share.html?TuSd">Show transmembrane proteins</a></b><br/>
If the protein is a transmembrane protein, you can click "File > Retrieve by ID > OPM PDB ID" to input a PDB ID to view the membranes.<br/>
<br/>
<li><b>Show <a href="https://structure.ncbi.nlm.nih.gov/icn3d/share.html?aYAjP4S3NbrBJX3x6">surface</a>, <a href="https://structure.ncbi.nlm.nih.gov/icn3d/share.html?L4C4WYE85tYRiFeK7">EM map</a>, or <a href="https://structure.ncbi.nlm.nih.gov/icn3d/share.html?QpqNZ3k65ToYFvUB6">electron density map</a></b><br/>
You can click "Style > Surface Type", "Style > EM Density Map", or "Style > Electron Density".<br/>
<br/>
<li><b>View 1D sequences and 2D interactions</b><br/>
In the page https://www.ncbi.nlm.nih.gov/Structure/icn3d/full.html?mmdbid=1TUP, you can click in "Windows" menu to "View Sequences & Annotations",
"View Interactions", and see all "Defined Sets", which can be clicked to see any of your selections.<br/>
<br/>
<li><b>Select on 3D, 1D and 2D</b><br/>
To select on 3D structures: hold "Alt" and use mouse to pick, hold "Ctrl" to union selection, hold "Shift" to select a range, press the up/down arrow to switch among atom/residue/strand/chain/structure. Click "Save Selection" in "Select" menu to save the current selection.<br/>
<br/>
To select on 1D sequences: drag on the sequences or the blue track title to select.<br/>
<br/>
To select on 2D interaction diagram: click on the nodes or lines. The nodes are chains and can be united with the Ctrl key. The lines are interactions and can NOT be united. Each click on the lines selects half of the lines, i.e., select the interacting residues in one of the two chains.<br/>
<br/>
<li><b><a href="https://structure.ncbi.nlm.nih.gov/icn3d/share.html?wPoW56e8QnzVfuZw6">Align two structures</a></b>, <b><a href="https://structure.ncbi.nlm.nih.gov/icn3d/share.html?ijnf">align two chains</a></b>, or <b><a href="https://structure.ncbi.nlm.nih.gov/icn3d/share.html?Mmm82craCwGMAxru9">align a protein sequence to a structure</a></b><br/>
You can click "File > Align" to see all three alignment options. You can also <a href="https://structure.ncbi.nlm.nih.gov/icn3d/share.html?tu166KaK8jQYUAVf7">realign</a> a subset of the structures.<br/>
<br/>
<li><b><a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/full.html?mmdbid=1tup&showanno=1&show2d=1&showsets=1&command=view+annotations;+set+view+detailed+view;+add+track+|+chainid+1TUP_B+|+title+Custom+Key+Sites+|+text+82+R,+152+G,+155-156+RR,+180+R,+189+R;+select+.B:82,152,155-156,180,189+|+name+mutation">Add custom tracks</a></b><br/>
You can add custom tracks in various formats (FASTA, bed file, etc.) in the annotation window by clicking the menu "Windows > View Sequences & Annotations".<br/>
<br/>

<li><b><a href="https://structure.ncbi.nlm.nih.gov/icn3d/share.html?rshvjTFXpAFu8GDa9">Show force-directed graph for interactions</a></b><br/>
You can show the interactions using 2D force-directed graph in the menu "View > H-Bonds & Interactions > 2D Graph (Force-Directed)".<br/>
<br/>

<li><b><a href="https://structure.ncbi.nlm.nih.gov/icn3d/share.html?xKSyfd1umbKstGh29">Calculate and show Solvent accessible surface area (SASA)</a></b><br/>
You can color structures with SASA, or show the SASA for each residue.<br/>
<br/>

<li><b><a href="https://structure.ncbi.nlm.nih.gov/icn3d/share.html?bGH1BfLsiGFhhTDn8">Show symmetry</a></b><br/>
You can show symmetry when available.<br/>
<br/>

</ol>

<li><b><a href="#HowToUse">How to embed iCn3D Structure Viewer in your html page?</a></b></li>
    <ul>
    <li><a href="#embedIframe">Embed using iframe</a></li>
    <li><a href="#embedJS">Embed using Javascript</a><br>
        <ol>
        <li><a href="#HowToUseStep1">Include three.js and jQuery javascript libraries in the &lt;head&gt; of your html page</a></li>
        <li><a href="#HowToUseStep2">Position the widget with an html div</a></li>
        <li><a href="#HowToUseStep3">Initialize the widget with desired parameters</a></li>
        </ol>
    </li>
    <li><a href="#modifyFeatures">Modify default features of iCn3D</a><br>
        <ul>
        <li><a href="#modifyOptions">Modify display options</a></li>
        <li><a href="#addUrlPara">Add URL parameters with or without commands</a></li>
        <li><a href="#modifyMethods">Modify iCn3D methods</a></li>
        </ul>
    <li><a href="#hints">Hints for using iCn3D</a><br>
        <ul>
        <li><a href="#HowToUseStep4">Hint for rotating/translating/zooming the 3D structure</a></li>
        <li><a href="#HowToUseStep5">Hint for selecting 3D structure and sequences</a></li>
        <!--li><a href="#HowToUseStep6">Hint for recovering a crashed page</a></li-->
        <li><a href="#HowToUseStep7">Hint for turning on scroll bars in Mac</a></li>
        </ul>
    </li>
    </ul>
<br>
<li><b><a href="#parameters">URL parameters for the advanced interface</a></b></li><br>
<li><b><a href="#commands">Menus, Commands, and Methods of the advanced interface</a></b>
    <table>
        <tbody><tr>
        <td>1. <a href="#file">File</a>&nbsp;&nbsp;</td>
        <td>2. <a href="#select">Select</a>&nbsp;&nbsp;</td>
        <td>2b. <a href="#selectb">Selection Definition</a>&nbsp;&nbsp;</td>
        <td>3. <a href="#view">View</a>&nbsp;&nbsp;</td>
        <td>4. <a href="#style">Style</a>&nbsp;&nbsp;</td>
        <td>5. <a href="#color">Color</a>&nbsp;&nbsp;</td>
        <td>6. <a href="#windows">Windows</a>&nbsp;&nbsp;</td>
        <td>7. <a href="#help">Help</a>&nbsp;&nbsp;</td>
        </tr>
        </tbody>
    </table>
</li><br>
<li><b><a href="#DataStructure">Data Structure</a></b></li><br>
<li><b><a href="#DisplayOptions">Display Options</a></b></li><br>
<li><b><a href="#Methods">Method Documents of the core library iCn3D</a></b></li>
<table>
<tbody><tr>
<td valign="top">
<div style="float:left; margin-left:10px;">
<span style="font-weight:bold; font-size:1em;">Options</span><br>
<a href="#bHighlight">bHighlight</a><br>
<a href="#bImpo">bImpo</a><br>
<a href="#bRender">bRender</a><br>
<a href="#bStopRotate">bStopRotate</a><br>
<a href="#coilWidth">coilWidth</a><br>
<a href="#cylinderHelixRadius">cylinderHelixRadius</a><br>
<a href="#dAtoms">dAtoms</a><br>
<a href="#dotSphereScale">dotSphereScale</a><br>
<a href="#hAtoms">hAtoms</a><br>
<a href="#hColor">hColor</a><br>
<a href="#helixSheetWidth">helixSheetWidth</a><br>
<a href="#lineRadius">lineRadius</a><br>
<a href="#matShader">matShader</a><br>
<a href="#nucleicAcidWidth">nucleicAcidWidth</a><br>
<a href="#pAtom">pAtom</a><br>
<a href="#pAtom2">pAtom2</a><br>
<a href="#pk">pk</a><br>
<a href="#ribbonthickness">ribbonthickness</a><br>
<a href="#sphereRadius">sphereRadius</a><br>
<a href="#stickRadius">stickRadius</a><br>
<a href="#traceRadius">traceRadius</a><br>
</div>

<div style="float:left;">
<span style="margin-left:10px; font-weight:bold; font-size:1em;">Methods</span><br>

<table><tbody><tr>
<td valign="top">
<div style="float:left; margin-left:10px;">
<a href="#addHlObjects">addHlObjects</a><br>
<a href="#applyDisplayOptions">applyDisplayOptions</a><br>
<a href="#applyOtherOptions">applyOtherOptions</a><br>
<a href="#applySsbondsOptions">applySsbondsOptions</a><br>
<a href="#applyTransformation">applyTransformation</a><br>
<a href="#buildAxes">buildAxes</a><br>
<a href="#calculateChemicalHbonds">calculateChemicalHbonds</a><br>
<a href="#centerAtoms">centerAtoms</a><br>
<a href="#createBox">createBox</a><br>
<a href="#createCylinderCurve">createCylinderCurve</a><br>
<a href="#createCylinderHelix">createCylinderHelix</a><br>
<a href="#createLabelRepresentation">createLabelRepresentation</a><br>
<a href="#createLineRepresentation">createLineRepresentation</a><br>
<a href="#createLines">createLines</a><br>
<a href="#createSphereRepresentation">createSphereRepresentation</a><br>
<a href="#createStickRepresentation">createStickRepresentation</a><br>
<a href="#createStrand">createStrand</a><br>
<a href="#createSurfaceRepresentation">createSurfaceRepresentation</a><br>
<a href="#createTube">createTube</a><br>
<a href="#draw">draw</a><br>
</div>

<div style="float:left; margin-left:10px;">
<a href="#drawCartoonNucleicAcid">drawCartoonNucleicAcid</a><br>
<a href="#drawNucleicAcidStick">drawNucleicAcidStick</a><br>
<a href="#drawSymmetryMates2">drawSymmetryMates2</a><br>
<a href="#exportCanvas">exportCanvas</a><br>
<a href="#getAtomsWithinAtom">getAtomsWithinAtom</a><br>
<a href="#getChainsFromAtoms">getChainsFromAtoms</a><br>
<a href="#getExtent">getExtent</a><br>
<a href="#getFirstAtomObj">getFirstAtomObj</a><br>
<a href="#getLastAtomObj">getLastAtomObj</a><br>
<a href="#getResiduesFromAtoms">getResiduesFromAtoms</a><br>
<a href="#hasCovalentBond">hasCovalentBond</a><br>
<a href="#init">init</a><br>
<a href="#loadPDB">loadPDB</a><br>
<a href="#rebuildScene">rebuildScene</a><br>
<a href="#reinitAfterLoad">reinitAfterLoad</a><br>
<a href="#render">render</a><br>
<a href="#removeHlObjects">removeHlObjects</a><br>
<a href="#removeSurfaces">removeSurfaces</a><br>
<a href="#residueName2Abbr">residueName2Abbr</a><br>
</div>

<div style="float:left; margin-left:10px;">
<a href="#rotateDown">rotateDown</a><br>
<a href="#rotateLeft">rotateLeft</a><br>
<a href="#rotateRight">rotateRight</a><br>
<a href="#rotateUp">rotateUp</a><br>
<a href="#saveFile">saveFile</a><br>
<a href="#selectStrandHelixFromAtom">selectStrandHelixFromAtom</a><br>
<a href="#setAtomStyleByOptions">setAtomStyleByOptions</a><br>
<a href="#setCamera">setCamera</a><br>
<a href="#setOutlineColor">setOutlineColor</a><br>
<a href="#setRotationCenter">setRotationCenter</a><br>
<a href="#setStyle2Atoms">setStyle2Atoms</a><br>
<a href="#setWidthHeight">setWidthHeight</a><br>
<a href="#showPicking">showPicking</a><br>
<a href="#translateDown">translateDown</a><br>
<a href="#translateLeft">translateLeft</a><br>
<a href="#translateRight">translateRight</a><br>
<a href="#translateUp">translateUp</a><br>
<a href="#zoomIn">zoomIn</a><br>
<a href="#zoomOut">zoomOut</a><br>
</div>

</td>

</tr></tbody></table>

</div>

<div style="float:left; margin-left:10px;">
<span style="font-weight:bold; font-size:1em;">Methods in Menus</span><br>
<a href="#addResiudeLabels">addResiudeLabels</a><br>
<a href="#applyEmmapOptions">applyEmmapOptions</a><br>
<a href="#applyMapOptions">applyMapOptions</a><br>
<a href="#applySurfaceOptions">applySurfaceOptions</a><br>
<a href="#centerSelection">centerSelection</a><br>
<a href="#hideHbonds">hideHbonds</a><br>
<a href="#resetOrientation">resetOrientation</a><br>
<a href="#setColorByOptions">setColorByOptions</a><br>
<a href="#zoominSelection">zoominSelection</a><br>
<br>
<span style="font-weight:bold; font-size:1em;">Hash Operations</span><br>
<a href="#cloneHash">cloneHash</a><br>
<a href="#exclHash">exclHash</a><br>
<a href="#exclHash2Atoms">exclHash2Atoms</a><br>
<a href="#hash2Atoms">hash2Atoms</a><br>
<a href="#intHash">intHash</a><br>
<a href="#intHash2Atoms">intHash2Atoms</a><br>
<a href="#unionHash">unionHash</a><br>
<a href="#unionHash2Atoms">unionHash2Atoms</a><br>
</div>

</td></tr>
</tbody></table>
<br>

<li><b><a href="#UIMethods">Method Documents of the advanced UI library iCn3DUI</a></b></li>
<table>
<tbody><tr>
<td valign="top">
<div style="float:left; margin-left:10px;">
<span style="font-weight:bold; font-size:1em;">Options</span><br>
<a href="#b2DShown">b2DShown</a><br>
<a href="#bAddCommands">bAddCommands</a><br>
<a href="#bAddLogs">bAddLogs</a><br>
<a href="#bAnnoShown">bAnnoShown</a><br>
<a href="#bNotLoadStructure">bNotLoadStructure</a><br>
<a href="#bSetChainsAdvancedMenu">bSetChainsAdvancedMenu</a><br>
<a href="#cfg">cfg</a><br>
<a href="#LESSWIDTH">LESSWIDTH</a><br>
<a href="#LESSHEIGHT">LESSHEIGHT</a><br>
<a href="#options">opts</a><br>
<a href="#pre">pre</a><br>

<br>
<a name="ModifyList"></a><span style="font-weight:bold; font-size:1em;">Modify iCn3DUI</span><br>
<a href="#modifyIcn3d">modifyIcn3d</a><br>
<a href="#allCustomEvents">allCustomEvents</a><br>
<a href="#setCustomDialogs">setCustomDialogs</a><br>

<br>
<span style="font-weight:bold; font-size:1em;">3D Print</span><br>
<a href="#hideStabilizer">hideStabilizer</a><br>
<a href="#prepareFor3Dprint">prepareFor3Dprint</a><br>
<a href="#resetAfter3Dprint">resetAfter3Dprint</a><br>
<a href="#saveStlFile">saveStlFile</a><br>
<a href="#saveVrmlFile">saveVrmlFile</a><br>
</div>

<div style="float:left; margin-left:10px;">
<span style="font-weight:bold; font-size:1em;">Methods in Menus</span><br>
<div style="float:left;">
<a href="#addChainLabels">addChainLabels</a><br>
<a href="#addLabel">addLabel</a><br>
<a href="#addLine">addLine</a><br>
<a href="#addStabilizer">addStabilizer</a><br>
<a href="#applyCommandAnnotationsAndCddSite">applyCommandAnnotationsAndCddSite</a><br>
<a href="#applyCommandEmmap">applyCommandEmmap</a><br>
<a href="#applyCommandMap">applyCommandMap</a><br>
<a href="#applySavedColor">applySavedColor</a><br>
<a href="#applySavedStyle">applySavedStyle</a><br>
<a href="#back">back</a><br>
<a href="#changeCustomAtoms">changeCustomAtoms</a><br>
<a href="#deleteSelectedSets">deleteSelectedSets</a><br>
<a href="#downloadAlignment">downloadAlignment</a><br>
<a href="#downloadBlast_rep_id">downloadBlast_rep_id</a><br>
<a href="#downloadCid">downloadCid</a><br>
<a href="#downloadGi">downloadGi</a><br>
<a href="#downloadMmcif">downloadMmcif</a><br>
<a href="#downloadMmdb">downloadMmdb</a><br>
<a href="#downloadMmcif">downloadMmcif</a><br>
<a href="#downloadPdb">downloadPdb</a><br>
<a href="#downloadUrl">downloadUrl</a><br>
<a href="#exportInteractions">exportInteractions</a><br>
<a href="#forward">forward</a><br>
<a href="#hideStabilizer">hideStabilizer</a><br>
<a href="#loadScript">loadScript</a><br>
<a href="#pickCustomSphere">pickCustomSphere</a><br>
<a href="#renderFinalStep">renderFinalStep</a><br>
<a href="#resetAfter3Dprint">resetAfter3Dprint</a><br>
<a href="#rotStruc">rotStruc</a><br>
<a href="#saveColor">saveColor</a><br>
<a href="#saveFile">saveFile</a><br>
<a href="#saveStyle">saveStyle</a><br>
<a href="#selectAll">selectAll</a><br>
</div>

<div style="float:left;">
<a href="#selectByCommand">selectByCommand</a><br>
<a href="#selectComplement">selectComplement</a><br>
<a href="#selectSideChains">selectSideChains</a><br>
<a href="#set2DDiagrams">set2DDiagrams</a><br>
<a href="#setOption">setOption</a><br>
<a href="#setStyle">setStyle</a><br>
<a href="#shareLink">shareLink</a><br>
<a href="#showHbonds">showHbonds</a><br>
<a href="#showSelection">showSelection</a><br>
<a href="#showSets">showSets</a><br>
<a href="#showSsbonds">showSsbonds</a><br>

<br>
<span style="font-weight:bold; font-size:1em;">Load Structures</span><br>
<a href="#loadAtomDataIn">loadAtomDataIn</a><br>
<a href="#loadSdfAtomData">loadSdfAtomData</a><br>
<a href="#loadMmcifData">loadMmcifData</a><br>
<a href="#loadPdbData">loadPdbData</a><br>

<br>
<span style="font-weight:bold; font-size:1em;">Display Structures</span><br>
<a href="#alternateStructures">alternateStructures</a><br>
<a href="#resizeCanvas">resizeCanvas</a><br>
<a href="#renderFinalStep">renderFinalStep</a><br><a href="#renderStructure">renderStructure</a><br>
<a href="#show3DStructure">show3DStructure</a><br>

<br>
<span style="font-weight:bold; font-size:1em;">Annotation Methods</span><br>
<a href="#showAnnotations">showAnnotations</a><br>
<a href="#showCddSite">showCddSite</a><br>
<a href="#showDomain">showDomain</a><br>
<a href="#showInteraction">showInteraction</a><br>
<a href="#showSeq">showSeq</a><br>
<a href="#showSnpClinvar">showSnpClinvar</a><br>
</div>

</div>

<div style="float:left; margin-left:10px;">
<span style="font-weight:bold; font-size:1em;">Selection Methods</span><br>
<a href="#addCustomSelection">addCustomSelection</a><br>
<a href="#applyCommand">applyCommand</a><br>
<a href="#applyCommandLoad">applyCommandLoad</a><br>
<a href="#clearSelection">clearSelection</a><br>
<a href="#execCommands">execCommands</a><br>
<a href="#outputSelection">outputSelection</a><br>
<a href="#selectAAlignChain">selectAAlignChain</a><br>
<a href="#selectAChain">selectAChain</a><br>
<a href="#switchHighlightLevelDown">switchHighlightLevelDown</a><br>
<a href="#switchHighlightLevelUp">switchHighlightLevelUp</a><br>
<a href="#toggleSelection">toggleSelection</a><br>
<br>
<span style="font-weight:bold; font-size:1em;">Highlight Methods</span><br>
<a href="#changeSeqColor">changeSeqColor</a><br>
<a href="#removeHl2D">removeHl2D</a><br>
<a href="#removeHlAll">removeHlAll</a><br>
<a href="#removeHlMenus">removeHlMenus</a><br>
<a href="#removeHlObjects">removeHlObjects</a><br>
<a href="#removeHlSeq">removeHlSeq</a><br>
<a href="#removeSeqChainBkgd">removeSeqChainBkgd</a><br>
<a href="#removeSeqResidueBkgd">removeSeqResidueBkgd</a><br>
<a href="#update2DdgmContent">update2DdgmContent</a><br>
<a href="#updateHl2D">updateHl2D</a><br>
<a href="#updateHlAll">updateHlAll</a><br>
<a href="#updateHlMenus">updateHlMenus</a><br>
<a href="#updateHlObjects">updateHlObjects</a><br>
<a href="#updateHlSeq">updateHlSeq</a><br>
</div>

<div style="float:left; margin-left:10px;">
<span style="font-weight:bold; font-size:1em;">Set Html</span><br>
<a href="#getAlignSequencesAnnotations">getAlignSequencesAnnotations</a><br>
<a href="#openDialog">openDialog</a><br>
<a href="#setAlignChainMenu">setAlignChainMenu</a><br>
<a href="#setAtomMenu">setAtomMenu</a><br>
<a href="#setDialogs">setDialogs</a><br>
<a href="#setLogCmd">setLogCmd</a><br>
<a href="#setLogWindow">setLogWindow</a><br>
<a href="#setMenu1">setMenu1</a><br>
<a href="#setMenu2">setMenu2</a><br>
<a href="#setMenu3">setMenu3</a><br>
<a href="#setMenu4">setMenu4</a><br>
<a href="#setMenu5">setMenu5</a><br>
<a href="#setMenu6">setMenu6</a><br>
<a href="#setTools">setTools</a><br>
<a href="#setTopMenusHtml">setTopMenusHtml</a><br>
<a href="#showMenu">showMenu</a><br>
<a href="#showTitle">showTitle</a><br>
<br>
<span style="font-weight:bold; font-size:1em;">Other Methods</span><br>
<a href="#allEventFunctions">allEventFunctions</a><br>
<a href="#getCommandsBeforeCrash">getCommandsBeforeCrash</a><br>
<a href="#hideMenu">hideMenu</a><br>
<a href="#initUI">init</a><br>
<a href="#isIE">isIE</a><br>
<a href="#isMac">isMac</a><br>
<a href="#isMobile">isMobile</a><br>
<a href="#isSessionStorageSupported">isSessionStorageSupported</a><br>
<a href="#modifyIcn3dshowPicking">modifyIcn3dshowPicking</a><br>
<a href="#saveCommandsToSession">saveCommandsToSession</a><br>
</div>

</td>
</tr>
</tbody></table>
<br>
<li><b><a href="#log">Change Log</a></b></li>

</ul>

<hr>

<!-- =========== END_TABLE_OF_CONTENTS ============= -->


<!-- =========== HOW_TO_USE_ICN3D_WIDGET_3_STEPS ============= -->

<br>

<a name="gallery"></a>

<a name="covid19-gallery"></a>
<h2><a href="https://www.ncbi.nlm.nih.gov/structure/?term=covid-19">COVID-19-related</a> gallery with live examples <img src="https://www.ncbi.nlm.nih.gov/Structure/IMG/spacer.gif" width="25" height="1" border="0"><a href="#Top"><img src="https://www.ncbi.nlm.nih.gov/Structure/IMG/arrowup_blue.gif" width="12" height="12" border="0" alt="back to top"></a></h2>

<div class="gallery">
<a href="https://structure.ncbi.nlm.nih.gov/icn3d/share.html?RcnHS2h8UMueak5A8" target="_blank">
<img style="height:300px" src ="https://www.ncbi.nlm.nih.gov/Structure/icn3d/gallery/6VSB-So5PM9t7PWbzN9JC9.png"><br>Cryo-EM structure of the 2019-nCoV spike <br>in the prefusion conformation (PDB 6VSB)
</a>
</div>

<div class="gallery">
<a href="https://structure.ncbi.nlm.nih.gov/icn3d/share.html?Gw1Da9w9jMXkNwMm7" target="_blank">
<img style="height:300px" src ="https://www.ncbi.nlm.nih.gov/Structure/icn3d/gallery/6M17-UPU5Fxryc9MbPQsC8.png"><br>2019-nCoV RBD/ACE2-B0AT1<br> complex (PDB 6M17)
</a>
</div>

<div class="gallery">
<a href="https://structure.ncbi.nlm.nih.gov/icn3d/share.html?pvSYXSfVZSKzapuZA" target="_blank">
<img style="height:300px" src ="https://www.ncbi.nlm.nih.gov/Structure/icn3d/gallery/6M0J-pvSYXSfVZSKzapuZA.png"><br>SARS-COV-2 Receptor Binding domain <br>interacting with ACE2 receptor (PDB 6M0J)
</a>
</div>

<div class="gallery">
<a href="https://structure.ncbi.nlm.nih.gov/icn3d/share.html?CuXYgGLCukDeUKnJ6" target="_blank">
<img style="height:300px" src ="https://www.ncbi.nlm.nih.gov/Structure/icn3d/gallery/6M0J-CuXYgGLCukDeUKnJ6-linegraph.png"><br>2D interaction graph<br>(PDB 6M0J)
</a>
</div>


<div class="gallery">
<a href="https://structure.ncbi.nlm.nih.gov/icn3d/share.html?u7gp4xS9rn4hahcLA" target="_blank">
<img style="height:300px" src ="https://www.ncbi.nlm.nih.gov/Structure/icn3d/gallery/6M0J-u7gp4xS9rn4hahcLA-linegraph-replay.png"><br>Replay each step of <br>2D interaction graph (PDB 6M0J)
</a>
</div>

<div class="gallery">
<a href="https://structure.ncbi.nlm.nih.gov/icn3d/share.html?1CPsLzoer5SpBoNo9" target="_blank">
<img style="height:300px" src ="https://www.ncbi.nlm.nih.gov/Structure/icn3d/gallery/6M0J-rK8kQrLYXvG6YqrY6-force.png"><br>2D interaction display, with force on X-axis,<br>at "View > H-Bonds & Interactions > 2D Graph (Force-Directed)" (PDB 6M0J)
</a>
</div>

<div class="gallery">
<a href="https://structure.ncbi.nlm.nih.gov/icn3d/share.html?eH1X8fDsQpPBeD1C6" target="_blank">
<img style="height:300px" src ="https://www.ncbi.nlm.nih.gov/Structure/icn3d/gallery/6M0J-eH1X8fDsQpPBeD1C6-force-circle.png"><br>2D interaction display, with force on Circle,<br>at "View > H-Bonds & Interactions > 2D Graph (Force-Directed)" (PDB 6M0J)
</a>
</div>

<div class="gallery">
<a href="https://structure.ncbi.nlm.nih.gov/icn3d/share.html?SvuVwqxKAKU8Vo6g6" target="_blank">
<img style="height:300px" src ="https://www.ncbi.nlm.nih.gov/Structure/icn3d/gallery/6W41-SvuVwqxKAKU8Vo6g6.png"><br>RBD interacting with <br>a patient antibody CR3022 (PDB 6W41)
</a>
</div>

<div class="gallery">
<a href="https://structure.ncbi.nlm.nih.gov/icn3d/share.html?s8kydLio6XnziC1L7" target="_blank">
<img style="height:300px" src ="https://www.ncbi.nlm.nih.gov/Structure/icn3d/gallery/6w41_2dd8-s8kydLio6XnziC1L7.png"><br>Comparing two antibodies against SARS-COV-1 vs. SARS-COV-2<br> epitopes on the virus Receptor Binding Domain (PDBs 6W41 and 2DD8)
</a>
</div>

<div class="gallery">
<a href="https://structure.ncbi.nlm.nih.gov/icn3d/share.html?BKYzBV6KXMvQL7sr9" target="_blank">
<img style="height:300px" src ="https://www.ncbi.nlm.nih.gov/Structure/icn3d/gallery/6m0j-msa-BKYzBV6KXMvQL7sr9.png"><br>Show Multiple Sequence Alignment<br> as Tracks and Add Custom Color (PDB 6M0J)
</a>
</div>

<div class="gallery">
<a href="https://structure.ncbi.nlm.nih.gov/icn3d/share.html?zDjCoXuxHE2PK1KMA" target="_blank">
<img style="height:300px" src ="https://www.ncbi.nlm.nih.gov/Structure/icn3d/gallery/6LU7-zDjCoXuxHE2PK1KMA.png"><br>COVID-19 main protease in complex<br> with an inhibitor N3 (PDB 6LU7)
</a>
</div>

<div style="clear:both;"></div>
<br>

<h2>iCn3D gallery with live examples<span style="font-size:0.7em"> ("iCn3D PNG Images", all images below except the first five snapshots, can be loaded into iCn3D by clicking "Open File > iCn3D PNG Image" in <a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/full.html" target="_blank">iCn3D File menu</a>)</span> <img src="https://www.ncbi.nlm.nih.gov/Structure/IMG/spacer.gif" width="25" height="1" border="0"><a href="#Top"><img src="https://www.ncbi.nlm.nih.gov/Structure/IMG/arrowup_blue.gif" width="12" height="12" border="0" alt="back to top"></a></h2>

<div class="gallery">
<a href="https://structure.ncbi.nlm.nih.gov/icn3d/share.html?wUuKpZDbK3Xr79EQ7" target="_blank">
<img style="height:300px" src ="https://www.ncbi.nlm.nih.gov/Structure/icn3d/gallery/1TUP-wUuKpZDbK3Xr79EQ7.png"><br>Synchronized 1D, 2D and 3D Displays (PDB 1TUP)
</a>
</div>

<div class="gallery">
<a href="https://structure.ncbi.nlm.nih.gov/icn3d/share.html?gVnT" target="_blank">
<img style="height:300px" src ="https://www.ncbi.nlm.nih.gov/Structure/icn3d/gallery/1TUP-mobilemenu-gVnT.png"><br>Mobile-style Menus (PDB 1TUP)
</a>
</div>

<div class="gallery">
<a href="https://structure.ncbi.nlm.nih.gov/icn3d/share.html?wPoW56e8QnzVfuZw6" target="_blank">
<img style="height:300px" src ="https://www.ncbi.nlm.nih.gov/Structure/icn3d/gallery/1HHO-4N7N-wPoW56e8QnzVfuZw6.png"><br>VAST+ Structure Alignment (PDBs 1HHO and 4N7N)
</a>
</div>

<div class="gallery">
<a href="https://structure.ncbi.nlm.nih.gov/icn3d/share.html?Mmm82craCwGMAxru9" target="_blank">
<img style="height:300px" src ="https://www.ncbi.nlm.nih.gov/Structure/icn3d/gallery/1TSR-align-Mmm82craCwGMAxru9.png"><br>Align Protein NP_001108451.1 to Structure 1TSR (PDB 1TSR)
</a>
</div>

<div class="gallery">
<a href="https://structure.ncbi.nlm.nih.gov/icn3d/share.html?AzdRqTaUscB74JSq8" target="_blank">
<img style="height:300px" src ="https://www.ncbi.nlm.nih.gov/Structure/icn3d/gallery/2ajf_6w41_side_by_side_h7bBXPkeu71C7GgR8.png"><br>Side-by-Side View (PDBs 2AJF and 6W41)
</a>
</div>

<div class="gallery">
<a href="https://structure.ncbi.nlm.nih.gov/icn3d/share.html?ijnf" target="_blank">
<img style="height:300px" src ="https://www.ncbi.nlm.nih.gov/Structure/icn3d/gallery/1HHO-4N7N-ijnf.png"><br>Align chain A of PDB 1HHO with chain A of PDB 4N7N
</a>
</div>

<div class="gallery">
<a href="https://structure.ncbi.nlm.nih.gov/icn3d/share.html?7g4Ae8TBpRQENAtS9" target="_blank">
<img style="height:300px" src ="https://www.ncbi.nlm.nih.gov/Structure/icn3d/gallery/1TUP-kQA2.png"><br>Hydrogen Bonds and Zn Interactions (PDB 1TUP)
</a>
</div>

<div class="gallery">
<a href="https://structure.ncbi.nlm.nih.gov/icn3d/share.html?JR5B" target="_blank">
<img style="height:300px" src ="https://www.ncbi.nlm.nih.gov/Structure/icn3d/gallery/3GVU-JR5B.png"><br>Ligand Binding with Fog and Slab (PDB 3GVU)
</a>
</div>

<div class="gallery">
<a href="https://structure.ncbi.nlm.nih.gov/icn3d/share.html?D6LC" target="_blank">
<img style="height:300px" src ="https://www.ncbi.nlm.nih.gov/Structure/icn3d/gallery/3GVU-D6LC.png"><br>Ligand Binding with Thinner Coils (PDB 3GVU)
</a>
</div>

<div class="gallery">
<a href="https://structure.ncbi.nlm.nih.gov/icn3d/share.html?vzTs" target="_blank">
<img style="height:300px" src ="https://www.ncbi.nlm.nih.gov/Structure/icn3d/gallery/3GVU-vzTs.png"><br>Ligand Binding with Surfaces and Labels (PDB 3GVU)
</a>
</div>

<div class="gallery">
<a href="https://structure.ncbi.nlm.nih.gov/icn3d/share.html?ead1" target="_blank">
<img style="height:300px" src ="https://www.ncbi.nlm.nih.gov/Structure/icn3d/gallery/6jxr-ead1.png"><br>Charged Residues in Transmembrane (PDB 6JXR)
</a>
</div>

<div class="gallery">
<a href="https://structure.ncbi.nlm.nih.gov/icn3d/share.html?xj2K" target="_blank">
<img style="height:300px" src ="https://www.ncbi.nlm.nih.gov/Structure/icn3d/gallery/6jxr-xj2K.png"><br>TCR-epsilon and TCR-delta Interactions (PDB 6JXR)
</a>
</div>

<div class="gallery">
<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/full.html" target="_blank">
<img style="height:300px" src ="https://www.ncbi.nlm.nih.gov/Structure/icn3d/gallery/custom_image_icn3d_loadable.png"><br>Custom PDB (load the image in iCn3D)
</a>
</div>

<div class="gallery">
<a href="https://structure.ncbi.nlm.nih.gov/icn3d/share.html?17g3r1JDvZ7ZL39e6" target="_blank">
<img style="height:300px" src ="https://www.ncbi.nlm.nih.gov/Structure/icn3d/gallery/2kod-17g3r1JDvZ7ZL39e6.png"><br>Multiple Structures (PDB 2KOD, press "a" to alternate)
</a>
</div>

<div class="gallery">
<a href="https://structure.ncbi.nlm.nih.gov/icn3d/share.html?gLx2qTdDVySqxcXv7" target="_blank">
<img style="height:300px" src ="https://www.ncbi.nlm.nih.gov/Structure/icn3d/gallery/3GVU-gLx2qTdDVySqxcXv7.png"><br>Electron Density Map (PDB 3GVU)
</a>
</div>

<div class="gallery">
<a href="https://structure.ncbi.nlm.nih.gov/icn3d/share.html?QpqNZ3k65ToYFvUB6" target="_blank">
<img style="height:300px" src ="https://www.ncbi.nlm.nih.gov/Structure/icn3d/gallery/3GVU-ed2-QpqNZ3k65ToYFvUB6.png"><br>Electron Density Map for a Subset (PDB 3GVU)
</a>
</div>

<div class="gallery">
<a href="https://structure.ncbi.nlm.nih.gov/icn3d/share.html?yyYxeJSiYbSWRkHCA" target="_blank">
<img style="height:300px" src ="https://www.ncbi.nlm.nih.gov/Structure/icn3d/gallery/6ENY-yyYxeJSiYbSWRkHCA.png"><br>EM Density Map (PDB 6ENY)
</a>
</div>

<div class="gallery">
<a href="https://structure.ncbi.nlm.nih.gov/icn3d/share.html?L4C4WYE85tYRiFeK7" target="_blank">
<img style="height:300px" src ="https://www.ncbi.nlm.nih.gov/Structure/icn3d/gallery/6ENY-em2-L4C4WYE85tYRiFeK7.png"><br>EM Density Map for a Subset (PDB 6ENY)
</a>
</div>

<div class="gallery">
<a href="https://structure.ncbi.nlm.nih.gov/icn3d/share.html?bGH1BfLsiGFhhTDn8" target="_blank">
<img style="height:300px" src ="https://www.ncbi.nlm.nih.gov/Structure/icn3d/gallery/1KQ2-bGH1BfLsiGFhhTDn8.png"><br>Protein Symmetry Display (PDB 1KQ2)
</a>
</div>

<div class="gallery">
<a href="https://structure.ncbi.nlm.nih.gov/icn3d/share.html?1yydeU3ktdkPyptAA" target="_blank">
<img style="height:300px" src ="https://www.ncbi.nlm.nih.gov/Structure/icn3d/gallery/1TUP-1yydeU3ktdkPyptAA.png"><br>3D Printing (PDB 1TUP)
</a>
</div>

<div class="gallery">
<a href="https://structure.ncbi.nlm.nih.gov/icn3d/share.html?ypAj" target="_blank">
<img style="height:300px" src ="https://www.ncbi.nlm.nih.gov/Structure/icn3d/gallery/1KQ2-ypAj.png"><br>Schematic Style (PDB 1KQ2)
</a>
</div>

<div class="gallery">
<a href="https://structure.ncbi.nlm.nih.gov/icn3d/share.html?aYAjP4S3NbrBJX3x6" target="_blank">
<img style="height:300px" src ="https://www.ncbi.nlm.nih.gov/Structure/icn3d/gallery/4UN3-o-aYAjP4S3NbrBJX3x6.png"><br>Solid Surface (PDB 4UN3)
</a>
</div>

<div class="gallery">
<a href="https://structure.ncbi.nlm.nih.gov/icn3d/share.html?MT8QssuYbfsSumFHA" target="_blank">
<img style="height:300px" src ="https://www.ncbi.nlm.nih.gov/Structure/icn3d/gallery/4UN3-t-MT8QssuYbfsSumFHA.png"><br>Transparent Surface (PDB 4UN3)
</a>
</div>

<div class="gallery">
<a href="https://structure.ncbi.nlm.nih.gov/icn3d/share.html?DFSLHuzN5T8GYdJSA" target="_blank">
<img style="height:300px" src ="https://www.ncbi.nlm.nih.gov/Structure/icn3d/gallery/2MLR-DFSLHuzN5T8GYdJSA.png"><br>Lipid Bilayer (PDB 2MLR)
</a>
</div>

<div class="gallery">
<a href="https://structure.ncbi.nlm.nih.gov/icn3d/share.html?ZLxYRrdZTFTg5fuF8" target="_blank">
<img style="height:300px" src ="https://www.ncbi.nlm.nih.gov/Structure/icn3d/gallery/1R09-ZLxYRrdZTFTg5fuF8.png"><br>Large Structure (PDB 1R09)
</a>
</div>

<div class="gallery">
<a href="https://structure.ncbi.nlm.nih.gov/icn3d/share.html?xPSLXHmCTYY4AMUE6" target="_blank">
<img style="height:300px" src ="https://www.ncbi.nlm.nih.gov/Structure/icn3d/gallery/2QZV-iPgmQqMQ6tKNfqvo6.png"><br>Large Structure (PDB 2QZV)
</a>
</div>

<div class="gallery">
<a href="https://structure.ncbi.nlm.nih.gov/icn3d/share.html?M2koemdfWbewV1ri8" target="_blank">
<img style="height:300px" src ="https://www.ncbi.nlm.nih.gov/Structure/icn3d/gallery/3J3Q-M2koemdfWbewV1ri8.png"><br>Large Structure (PDB 3J3Q)
</a>
</div>

<div class="gallery">
<a href="https://structure.ncbi.nlm.nih.gov/icn3d/share.html?e9tDyp13ePPjnvdH6" target="_blank">
<img style="height:300px" src ="https://www.ncbi.nlm.nih.gov/Structure/icn3d/gallery/cid2244-all-e9tDyp13ePPjnvdH6.png"><br>Compound with Hydrogens (CID 2244)
</a>
</div>

<div class="gallery">
<a href="https://structure.ncbi.nlm.nih.gov/icn3d/share.html?y4t5iA1YvEoib2517" target="_blank">
<img style="height:300px" src ="https://www.ncbi.nlm.nih.gov/Structure/icn3d/gallery/cid2244-y4t5iA1YvEoib2517.png"><br>Compound without Hydrogens (CID 2244)
</a>
</div>

<div style="clear:both;"></div>
<br>

<hr>

<a name="HowToUse"></a>
<h2>How to embed iCn3D Structure Viewer in your html page:<img src="https://www.ncbi.nlm.nih.gov/Structure/IMG/spacer.gif" width="25" height="1" border="0"><a href="#Top"><img src="https://www.ncbi.nlm.nih.gov/Structure/IMG/arrowup_blue.gif" width="12" height="12" border="0" alt="back to top"></a></h2>
<ul>
<li><a name="embedIframe"></a><b>Embed using iframe</b><br>
<p>To embed iCn3D Structure Viewer in your own HTML page using iframe, simply add the following to your page:
</p><div class="indent"><code>&lt;iframe src="https://www.ncbi.nlm.nih.gov/Structure/icn3d/full.html?mmdbid=1tup&amp;width=300&amp;height=300&amp;showcommand=0&amp;mobilemenu=1&amp;showtitle=0" width="320" height="320" style="border:none"&gt;&lt;/iframe&gt;</code></div><br>

</li>
<li><a name="embedJS"></a><b>Embed using Javascript</b><br>

<p>To embed iCn3D Structure Viewer in your own HTML page using the distribution files on NCBI servers, simply follow these 3 steps: (To embed multiple iCn3D viewers, see the source code of the <a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/example.html">example page</a> for reference)</p>

<ol>
    <li><a name="HowToUseStep1"></a><b>Include the CSS and JavaScript libraries in the &lt;head&gt; of your html page, as shown here.</b></li>
    <ul><li>
      <p>Basic UI:</p>
    <pre>&lt;link rel="stylesheet" href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/lib/jquery-ui.min.css"&gt;
&lt;link rel="stylesheet" href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d_simple_ui.css"&gt;
&lt;script src="https://www.ncbi.nlm.nih.gov/Structure/icn3d/lib/jquery.min.js"&gt;&lt;/script&gt;
&lt;script src="https://www.ncbi.nlm.nih.gov/Structure/icn3d/lib/jquery-ui.min.js"&gt;&lt;/script&gt;
&lt;script src="https://www.ncbi.nlm.nih.gov/Structure/icn3d/lib/three.min.js"&gt;&lt;/script&gt;
&lt;script src="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d_simple_ui.min.js"&gt;&lt;/script&gt;</pre>
    </li>
    <li><p>Advanced UI:</p>
    <pre>&lt;link rel="stylesheet" href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/lib/jquery-ui.min.css"&gt;
&lt;link rel="stylesheet" href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d_full_ui.css"&gt;
&lt;script src="https://www.ncbi.nlm.nih.gov/Structure/icn3d/lib/jquery.min.js"&gt;&lt;/script&gt;
&lt;script src="https://www.ncbi.nlm.nih.gov/Structure/icn3d/lib/jquery-ui.min.js"&gt;&lt;/script&gt;
&lt;script src="https://www.ncbi.nlm.nih.gov/Structure/icn3d/lib/three.min.js"&gt;&lt;/script&gt;
&lt;script src="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d_full_ui.min.js"&gt;&lt;/script&gt;</pre>
    </li>
    </ul>

    <li><a name="HowToUseStep2"></a><b>Position the widget with an html div</b></li>
    <pre>      &lt;div id="icn3dwrap"&gt;&lt;/div&gt;
    </pre>
    The widget will be rendered inside of the div.<br><br>


    <li><a name="HowToUseStep3"></a><b>Initialize the widget with desired parameters:</b></li>
    <ul><li>Basic UI:<br>
    <pre>  &lt;script type="text/javascript"&gt;
    $( document ).ready(function() {
      var cfg = {
          divid: 'icn3dwrap',
          width: 400,
          height: 400,
          resize: false,
          rotate: 'right',
          pdbid: '1tup'
      };

      var icn3dui = new iCn3DUI(cfg);
      icn3dui.show3DStructure();
    });&lt;/script&gt;
    </pre>
    </li>
    <li>Advanced UI:<br>
    <pre>  &lt;script type="text/javascript"&gt;
    $( document ).ready(function() {
          var options = {};

          //Options are available at: https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d.html#DisplayOptions
          //options['proteins'] = 'sphere';

          var cfg = {
              divid: 'icn3dwrap',
              width: '100%',
              height: '100%',
              resize: true,
              rotate: 'right',
              mobilemenu: true,
              showcommand: false,
              showtitle: false
          };
          cfg['pdbid'] = '1tup';
          if(Object.keys(options).length > 0) cfg['options'] = options;

          var icn3dui = new iCn3DUI(cfg);

          //communicate with the 3D viewer with chained functions
          $.when(icn3dui.show3DStructure()).then(function() {
               //icn3dui.setOption('color', 'cyan');
          });
    });
  &lt;/script&gt;</pre>
    </li>
    </ul>
</ol>
</li>

<li><a name="modifyFeatures"></a><b>Modify default features of iCn3D</b><br>
    <ul>
    <li><a name="modifyOptions"></a><b>Modify <a href="#DisplayOptions">display options</a>:</b></li>
        <div>Users can add display options in the file full.html, e.g., "options['proteins'] = 'sphere';".</div><br>
    <li><a name="addUrlPara"></a><b>Add <a href="#parameters">URL parameters</a> with or without commands:</b></li>
        <div>Users can add URL parameters to open the annotation window (&showanno=1), open the 2D interaction diagram (&show2d=1), open the defined sets (&showsets=1), etc. Users can also use the command parameter to apply commands on the structure, e.g., "&command=style protein sphere".</div><br>
    <li><a name="modifyMethods"></a><b>Modify iCn3D methods:</b></li>
        <div>Users can add custom methods by overwriting <a href="#ModifyList">these methods</a>: modifyIcn3d, allCustomEvents, setCustomDialogs</div><br>
    </ul>
</li>

<li><a name="hints"></a><b>Hints for using iCn3D</b><br>
  <ul>
    <li><a name="HowToUseStep4"></a><b>Hint for rotating/translating/zooming the 3D structure:</b></li>
    <ul>
      <li>Rotate
          <ul>
              <li>Left mouse</li>
              <li>Key L: left</li>
              <li>Key J: right</li>
              <li>Key I: up</li>
              <li>Key M: down</li>
              <li>Shift + Key L: left 90&deg;</li>
              <li>Shift + Key J: right 90&deg;</li>
              <li>Shift + Key I: up 90&deg;</li>
              <li>Shift + Key M: down 90&deg;</li>
          </ul>
      </li>
      <li>Zoom
          <ul>
              <li>Middle mouse</li>
              <li>Key Z: zoom in</li>
              <li>Key X: zoom out</li>
          </ul>
      </li>
      <li>Translate
          <ul>
              <li>Right mouse</li>
              <!--li>Arrow left: left</li>
              <li>Arrow right: right</li>
              <li>Arrow up: up</li>
              <li>Arrow down: down</li-->
          </ul>
      </li>
    </ul>
    <br>

    <li><a name="HowToUseStep5"></a><b>Hint for selcting 3D structure, 2D interactions, and 1D sequences:</b></li>
    <div>To pick an atom/residue/strand in the 3D structure, hold "Alt" key and use mouse to pick. Click the second time to deselect. To add more selection to the current selection, hold "Ctrl" key and use mouse to pick. To add a range of selection starting from the current selection, hold "Shift" key and use mouse to pick the ending selection. To change the selected residue to the strand/chain/structure containing the residue, press the up/down arrow. <br/>
    <br/>
    To select chains or interactions in the 2D interactions diagram, click on the nodes or lines. The nodes are chains and can be united with the Ctrl key. The lines are interactions and can NOT be united. Each click on the lines selects half of the lines, i.e., select the interacting residues in one of the two chains. The selected residues are saved in the "Select -> Advanced" menu.<br/>
    <br/>
    To select residues in the 1D sequence window, drag with mouse or touch to select residues. Drag or touch again to deselect residues. Multiple selection is allowed without the Ctrl key.<br/>
    <br/>
    To save the current selection (either on 3D structure, 2D interactions, or 1D sequence), open the menu "Select -> Save Selection", specify the name and description for the selection, and click "Save".
    </div>
    <br>

    <!--li><a name="HowToUseStep6"></a><b>Hint for recovering a crashed page:</b></li>
    <div>If for some reason the 3D viewer crashes, you can refresh the same crashed page to get the saved commands and recover your previous state by loading the commands at File -&gt; Open State/Script. This is done by using session storage. It works in Chrome, Firefox, and Internet Explorer in PC, but not Safari nor Mac.
    </div>
    <br-->

    <li><a name="HowToUseStep7"></a><b>Hint for turning on scroll bars in Mac:</b></li>
    <div>If you are using Macintosh computer and need to scroll in the sequence dialog, you can turn on the scroll bars as follows: Systems preferences -> GENERAL -> show scroll bars -> check "always".
    </div>

</ul>
</li>
</ul>

<!-- =========== END_HOW_TO_USE_ICN3D_WIDGET_3_STEPS ============= -->

<hr>

<a name="parameters"></a>
<h2>URL parameters for the advanced interface<img src="https://www.ncbi.nlm.nih.gov/Structure/IMG/spacer.gif" width="25" height="1" border="0"><a href="#Top"><img src="https://www.ncbi.nlm.nih.gov/Structure/IMG/arrowup_blue.gif" width="12" height="12" border="0" alt="back to top"></a></h2>
<table><tr valign='top'><th>Parameter</th><th>Basic interface<br/>(index.html)</th><th>Advanced interface<br/>(full.html)</th><th>Description</th></tr>
<tr valign='top'><td>mmdbid</td><td>Yes</td><td>Yes</td><td>NCBI MMDB ID or PDB ID, e.g., <a href="full.html?mmdbid=1tup&showanno=1&show2d=1">?mmdbid=1tup&showanno=1&show2d=1</a></td></tr>
<tr valign='top'><td>mmtfid</td><td>Yes</td><td>Yes</td><td>MMTF ID, e.g., <a href="full.html?mmtfid=1tup">?mmtfid=1tup</a></td></tr>
<tr valign='top'><td>pdbid</td><td>Yes</td><td>Yes</td><td>PDB ID, e.g., <a href="full.html?pdbid=1tup">?pdbid=1tup</a></td></tr>
<tr valign='top'><td>mmcifid</td><td>Yes</td><td>Yes</td><td>mmCIF ID, e.g., <a href="full.html?mmcifid=1tup">?mmcifid=1tup</a></td></tr>
<tr valign='top'><td>gi</td><td>Yes</td><td>Yes</td><td>NCBI protein gi number, e.g., <a href="full.html?gi=827343227">?gi=827343227</a></td></tr>
<tr valign='top'><td>cid</td><td>Yes</td><td>Yes</td><td>PubChem Compound ID, e.g., <a href="full.html?cid=2244">?cid=2244</a></td></tr>
<tr valign='top'><td>blast_rep_id</td><td>No</td><td>Yes</td><td>NCBI protein accession of a chain of a 3D structure. "blast_rep_id" combines with the parameter "query_id" of a protein sequence to show the sequence-structure alignment, e.g., <a href="full.html?from=icn3d&blast_rep_id=1TSR_A&query_id=NP_001108451.1">from=icn3d&blast_rep_id=1TSR_A&query_id=NP_001108451.1</a></td></tr>
<tr valign='top'><td>align</td><td>No</td><td>Yes</td><td>Two PDB IDs or MMDB IDs for structure alignment, e.g., <a href="full.html?align=1hho,4n7n">?align=1hho,4n7n</a></td></tr>
<tr valign='top'><td>chainalign</td><td>No</td><td>Yes</td><td>Two chains for structure alignment, e.g., <a href="full.html?chainalign=1HHO_A,4N7N_A">?chainalign=1HHO_A,4N7N_A</a></td></tr>
<tr valign='top'><td>url</td><td>Yes</td><td>Yes</td><td>Use the url (encoded) to retrieve the 3D structure. The url requires another parameter "type", e.g., <a href="full.html?type=pdb&url=https%3A%2F%2Ffiles.rcsb.org%2Fview%2F1gpk.pdb">?type=pdb&url=https%3A%2F%2Ffiles.rcsb.org%2Fview%2F1gpk.pdb</a></td></tr>

<tr valign='top'><td>width</td><td>Yes</td><td>Yes</td><td>Width of the structure image. It can be percentage such as '100%', or pixel values such as 400. The default is "100%".</td></tr>
<tr valign='top'><td>height</td><td>Yes</td><td>Yes</td><td>Height of the structure image. It can be percentage such as '100%', or pixel values such as 400. The default is "100%".</td></tr>
<tr valign='top'><td>resize</td><td>Yes</td><td>Yes</td><td>Set "true" or "1" to resize the image when the container is resized. The default is "true".</td></tr>
<tr valign='top'><td>rotate</td><td>Yes</td><td>Yes</td><td>Set "right", "left", "up", or "down" to rotate the structure when it is  displayed at the beginning. The default is "right".</td></tr>

<tr valign='top'><td>showanno</td><td>No</td><td>Yes</td><td>Set "true" or "1" to show annotations, such as SNPs, ClinVar, domains, binding sites. The default is "false".</td></tr>
<tr valign='top'><td>showalignseq</td><td>No</td><td>Yes</td><td>Set "true" or "1" to show the aligned sequence window. The default is "false".</td></tr>
<tr valign='top'><td>showsets</td><td>No</td><td>Yes</td><td>Set "true" or "1" to show the defined sets. The default is "false".</td></tr>
<tr valign='top'><td>show2d</td><td>No</td><td>Yes</td><td>Set "true" or "1" to show the 2D interaction. The default is "false".</td></tr>
<tr valign='top'><td>showmenu</td><td>Yes</td><td>Yes</td><td>Set "false" or "0" to hide the menus and buttons at the top of the structure canvas. The default is "true".</td></tr>
<tr valign='top'><td>showtitle</td><td>Yes</td><td>Yes</td><td>Set "false" or "0" to hide the title at the top of the structure canvas. The default is "true".</td></tr>
<tr valign='top'><td>showcommand</td><td>No</td><td>Yes</td><td>Set "false" or "0" to hide the command window. The default is "true".</td></tr>
<tr valign='top'><td>simplemenu</td><td>No</td><td>Yes</td><td>Set "true" or "1" to show only the menus "File", "View", "Style", and "Color" to simplify the UI. The default is "false".</td></tr>
<tr valign='top'><td>mobilemenu</td><td>No</td><td>Yes</td><td>Set "true" or "1" to show the mobile-style menu. Users can click to see all menus. The default is "false".</td></tr>
<tr valign='top'><td>closepopup</td><td>No</td><td>Yes</td><td>Set "true" or "1" to close the dialogs of "Defined Sets", "Interactions", and "Sequences and Annotations". The default is "false".</td></tr>
<tr valign='top'><td><a name="commandsUrl"></a>command</td><td>No</td><td>Yes</td><td>Besides the "Script/Log" section at the bottom of the canvas, commands can also be attached to the url parameter "command". The easy way to generate the commands is to operate manually to get your custom display and then click "File > Share Link" to find the URL with full commands or just the short URL.</td></tr>
<tr valign='top'><td>replay</td><td>No</td><td>Yes</td><td>Set "true" or "1" to replay each step of a custom display.</td></tr>
</table>

<hr>

<a name="commands"></a>
<h2>Menus, Commands, and Methods of the advanced interface<img src="https://www.ncbi.nlm.nih.gov/Structure/IMG/spacer.gif" width="25" height="1" border="0"><a href="#Top"><img src="https://www.ncbi.nlm.nih.gov/Structure/IMG/arrowup_blue.gif" width="12" height="12" border="0" alt="back to top"></a></h2>
<a name="file"></a><b>1. File</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
<table width='100%'><tr><td valign='top' width='30%'><b>Menu</b></td><td valign='top' width='40%'><b>Command</b></td><td valign='top'><b>Method</b></td></tr>
<tbody>
<tr><td colspan="3">a) Retrieve by ID</td></tr>
<tr><td valign='top'><span class="indent">MMDB ID:</span></td>          <td valign='top'>load mmdb [1GPK]</td> <td valign='top'><a href="#downloadMmdb">downloadMmdb</a></td></tr>
<tr><td width="240"><span class="indent">MMTF ID:</span></td>           <td valign='top'>load mmtf [1GPK]</td> <td valign='top'><a href="#downloadMmtf">downloadMmtf</a></td></tr>
<tr><td valign='top'><span class="indent">PDB ID:</span></td>           <td valign='top'>load pdb [1GPK]</td> <td valign='top'><a href="#downloadPdb">downloadPdb</a></td></tr>
<tr><td valign='top'><span class="indent">OPM PDB ID:</span></td>           <td valign='top'>load opm [6JXR]</td> <td valign='top'><a href="#downloadPdb">downloadOpm</a></td></tr>
<tr><td valign='top'><span class="indent">mmCIF ID:</span></td>          <td valign='top'>load mmcif [1GPK]</td> <td valign='top'><a href="#downloadMmcif">downloadMmcif</a></td></tr>
<tr><td valign='top'><span class="indent">NCBI gi:</span></td>               <td valign='top'>load gi [827343227]</td> <td valign='top'><a href="#downloadGi">downloadGi</a></td></tr>
<tr><td valign='top'><span class="indent">PubChem CID:</span></td>              <td valign='top'>load cid [2244]</td> <td valign='top'><a href="#downloadCid">downloadCid</a></td></tr>

<tr><td colspan="3">b) Open File</td> </tr>
<tr><td valign='top'><span class="indent">PDB File:</span></td>            <td valign='top'>[no command available]</td> <td valign='top'></td></tr>
<tr><td valign='top'><span class="indent">mmCIF File:</span></td>          <td valign='top'>[no command available]</td> <td valign='top'></td></tr>
<tr><td valign='top'><span class="indent">Mol2 File:</span></td>          <td valign='top'>[no command available]</td> <td valign='top'></td></tr>
<tr><td valign='top'><span class="indent">SDF File:</span></td>          <td valign='top'>[no command available]</td> <td valign='top'></td></tr>
<tr><td valign='top'><span class="indent">XYZ File:</span></td>          <td valign='top'>[no command available]</td> <td valign='top'></td></tr>
<tr><td valign='top'><span class="indent">URL (Same Host):</span></td>          <td valign='top'>load url [https://files.rcsb.org/view/1gpk.pdb]</td> <td valign='top'><a href="#downloadUrl">downloadUrl</a></td></tr>
<tr><td valign='top'><span class="indent">iCn3D PNG Image:</span></td>   <td valign='top'>[no command available]</td> <td valign='top'></td></tr>
<tr><td valign='top'><span class="indent">State/Script File:</span></td>   <td valign='top'>[no command available]</td> <td valign='top'><a href="#loadScript">loadScript</a></td></tr>
<tr><td valign='top'><span class="indent">Selection File:</span></td>   <td valign='top'>[no command available]</td> <td valign='top'></td></tr>
<tr><td valign='top'><span class="indent">Electron Density (DSN6) - Local File:</span></td>   <td valign='top'>set map 2fofc sigma 1.5</td> <td valign='top'>loadDsn6Data()</td></tr>
<tr><td valign='top'><span class="indent">Electron Density (DSN6) - URL (Same Host):</span></td>   <td valign='top'>set map 2fofc sigma 1.5 | [URL]</td> <td valign='top'>Dsn6ParserBase()</td></tr>

<tr><td colspan="3">c) Align</td> </tr>
<tr><td valign='top'><span class="indent">Sequence to Structure:</span></td>            <td valign='top'>load seq_struc_ids [23491729,1TUP_A]</td> <td valign='top'><a href="#downloadBlast_rep_id">downloadBlast_rep_id</a></td></tr>
<tr><td valign='top'><span class="indent">Structure to Structure - Invariant Substructure Superposed:</span></td>          <td valign='top'>load alignment [1HHO,4N7N] | parameters atype=1</td> <td valign='top'><a href="#downloadAlignment">downloadAlignment</a></td></tr>
<tr><td valign='top'><span class="indent">Structure to Structure - All Matching Molecules Superposed:</span></td>          <td valign='top'>load alignment [1HHO,4N7N] | parameters atype=0</td> <td valign='top'><a href="#downloadAlignment">downloadAlignment</a></td></tr>
<tr><td valign='top'><span class="indent">Chain to Chain:</span></td>          <td valign='top'>load chainalignment [1HHO_A,4N7N_A]</td> <td valign='top'><a href="#downloadChainAlignment">downloadChainAlignment</a></td></tr>

<tr><td valign='top'>d) Realign Selection</td>         <td valign='top'>realign on seq align</td> <td valign='top'><a href="#shareLink">realignOnSeqAlign</a></td></tr>

<tr><td colspan="3">e) 3D Printing</td> </tr>
<tr><td valign='top'><span class="indent">VRML (Color, W/ Stabilizers):</span></td>          <td valign='top'>export vrml stabilizer file</td> <td valign='top'><a href="#saveVrmlFile">saveVrmlFile</a></td></tr>
<tr><td valign='top'><span class="indent">STL (W/ Stabilizers):</span></td>          <td valign='top'>export stl stabilizer file</td> <td valign='top'><a href="#saveStlFile">saveStlFile</a></td></tr>

<tr><td valign='top'><span class="indent">VRML (Color):</span></td>          <td valign='top'>export vrml file</td> <td valign='top'><a href="#saveVrmlFile">saveVrmlFile</a></td></tr>
<tr><td valign='top'><span class="indent">STL:</span></td>          <td valign='top'>export stl file</td> <td valign='top'><a href="#saveStlFile">saveStlFile</a></td></tr>
<tr><td valign='top'><span class="indent">Add All Stabilizers:</span></td>          <td valign='top'>stabilizer</td> <td valign='top'><a href="#addStabilizer">addStabilizer</a></td></tr>
<tr><td valign='top'><span class="indent">Remove All Stabilizers:</span></td>          <td valign='top'>set stabilizer off</td> <td valign='top'><a href="#hideStabilizer">hideStabilizer</a></td></tr>
<tr><td valign='top'><span class="indent">Add One Stabilizer:</span></td>          <td valign='top'>add one stabilizer | [atom serial] [atom serial]</td> <td valign='top'></td></tr>
<tr><td valign='top'><span class="indent">Remove One Stabilizer:</span></td>          <td valign='top'>remove one stabilizer | [atom serial] [atom serial]</td> <td valign='top'></td></tr>
<tr><td valign='top'><span class="indent">Set Thickness:</span></td>          <td valign='top'>set thickness | linerad 0.6 | coilrad 1 | stickrad 0.8 | ribbonthick 0.8 | proteinwidth 2 | nucleotidewidth 1.2 | ballscale 0.6</td> <td valign='top'></td></tr>
<tr><td valign='top'><span class="indent">Reset Thickness:</span></td>          <td valign='top'>reset thickness</td> <td valign='top'><a href="#resetAfter3Dprint">resetAfter3Dprint</a></td></tr>

<tr><td colspan="3">f) Save Files</td> </tr>
<tr><td valign='top'><span class="indent">iCn3D PNG Image:</span></td>          <td valign='top'>export canvas (URL example: <a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/full.html?mmdbid=1top&command=export%20canvas" target="_blank">https://www.ncbi.nlm.nih.gov/Structure/icn3d/full.html?mmdbid=1top&command=export%20canvas</a>)</td> <td valign='top'><a href="#saveFile">saveFile([filename], 'png')</a></td></tr>
<tr><td valign='top'><span class="indent">State File:</span></td>            <td valign='top'>[no command available]</td> <td valign='top'><a href="#saveFile">saveFile</a></td></tr>
<tr><td valign='top'><span class="indent">All Selections:</span></td>   <td valign='top'>[no command available]</td> <td valign='top'><a href="#saveFile">saveFile</a></td></tr>
<tr><td valign='top'><span class="indent">Residue Counts:</span></td>   <td valign='top'>[no command available]</td> <td valign='top'><a href="#saveFile">saveFile</a></td></tr>
</tr>

<tr><td valign='top'>g) Share Link:</td>         <td valign='top'>[no command available]</td> <td valign='top'><a href="#shareLink">shareLink</a></td></tr>

<tr><td valign='top'>g) Replay Each Step:</td>         <td valign='top'>replay on</td> <td valign='top'>replayon</td></tr>

</tbody></table>
<br>
<a name="select"></a><b>2. Select</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
<table width='100%'><tr><td valign='top' width='30%'><b>Menu</b></td><td valign='top' width='40%'><b>Command</b></td><td valign='top'><b>Method</b></td></tr>
<tr><td valign='top'>a) Defined Sets:</td>     <td valign='top'>defined sets</td> <td valign='top'><a href="#showSets">showSets</a></td></tr>
<tr><td valign='top'><span class="indent">Select sets:</span></td>    <td valign='top'>select saved atoms [name1] or/and/not [name2]</td> <td valign='top'><a href="#changeCustomAtoms">changeCustomAtoms</a></td></tr>
<tr><td valign='top'><span class="indent">Delete selected sets:</span></td>    <td valign='top'>delete selected sets</td> <td valign='top'><a href="#deleteSelectedSets">deleteSelectedSets</a></td></tr>

<tr><td valign='top'>b) All:</td>     <td valign='top'>select all</td> <td valign='top'><a href="#selectAll">selectAll</a></td></tr>
<tr><td valign='top'>c) Displayed Set:</td>     <td valign='top'>select displayed set</td> <td valign='top'>me.icn3d.hAtoms = me.icn3d.cloneHash(me.icn3d.dAtoms)</td></tr>

<tr><td colspan="3">d) by Distance</td> </tr>
<tr><td valign='top'><span class="indent">Display:</span></td>            <td valign='top'>select zone cutoff [cutoff in angstrom]</td> <td valign='top'><a href="#pickCustomSphere">pickCustomSphere</a></td></tr>

<tr><td colspan="3"><br>e) by Property</td> </tr>
<tr><td valign='top'><span class="indent">Positive:</span></td>  <td valign='top'>select prop positive</td> <td valign='top'>selectProperty</td></tr>
<tr><td valign='top'><span class="indent">Negative:</span></td>  <td valign='top'>select prop negative</td> <td valign='top'>selectProperty</td></tr>
<tr><td valign='top'><span class="indent">Hydrophobic:</span></td>  <td valign='top'>select prop hydrophobic</td> <td valign='top'>selectProperty</td></tr>
<tr><td valign='top'><span class="indent">Polar:</span></td>  <td valign='top'>select prop polar</td> <td valign='top'>selectProperty</td></tr>
<tr><td valign='top'><span class="indent">B-factor:</span></td>  <td valign='top'>select prop b factor</td> <td valign='top'>selectProperty</td></tr>
<tr><td valign='top'><span class="indent">Solvent Accessibility:</span></td>  <td valign='top'>select prop percent out</td> <td valign='top'>selectProperty</td></tr>

<tr><td valign='top'>f) Inverse:</td>     <td valign='top'>select complement</td> <td valign='top'><a href="#selectComplement">selectComplement</a></td></tr>
<tr><td valign='top'>g) Main Chains:</td>     <td valign='top'>select main chains</td> <td valign='top'><a href="#selectMainChains">selectMainChains</a></td></tr>
<tr><td valign='top'>h) Side Chains:</td>     <td valign='top'>select side chains</td> <td valign='top'><a href="#selectSideChains">selectSideChains</a></td></tr>

<tr><td valign='top'>i) Advanced</td><td valign='top'>select [command] | name [commandname]</td> <td valign='top'><a href="#selectByCommand">selectByCommand</a></td></tr>

<tr><td colspan="3"><br>j) Select on 3D</td> </tr>
<tr><td valign='top'><span class="indent">Chain (Alt):</span></td>  <td valign='top'>set pk chain</td> <td valign='top'>me.icn3d.pk = 5; me.icn3d.opts['pk'] = 'chain';</td></tr>
<tr><td valign='top'><span class="indent">3D Domain (Alt):</span></td>  <td valign='top'>set pk domain</td> <td valign='top'>me.icn3d.pk = 4; me.icn3d.opts['pk'] = 'domain';</td></tr>
<tr><td valign='top'><span class="indent">Strand/Helix (Alt):</span></td>  <td valign='top'>set pk strand</td> <td valign='top'>me.icn3d.pk = 3; me.icn3d.opts['pk'] = 'strand';</td></tr>
<tr><td valign='top'><span class="indent">Residue (Alt):</span></td>  <td valign='top'>set pk residue</td> <td valign='top'>me.icn3d.pk = 2; me.icn3d.opts['pk'] = 'residue';</td></tr>
<tr><td valign='top'><span class="indent">Atom (Alt):</span></td>     <td valign='top'>set pk atom</td> <td valign='top'>me.icn3d.pk = 1; me.icn3d.opts['pk'] = 'atom';</td></tr>

<tr><td valign='top'>k) Save Selection:</td>      <td valign='top'>select [command] | name [commandname]</td> <td valign='top'><a href="#selectByCommand">selectByCommand</a></td></tr>
<tr><td valign='top'>l) Clear Selection:</td>      <td valign='top'>clear all</td> <td valign='top'><a href="#selectAll">selectAll</a></td></tr>

<tr><td colspan="3"><br>m) Highlight Color</td> </tr>
<tr><td valign='top'><span class="indent">Yellow:</span></td>   <td valign='top'>set highlight color yellow</td> <td valign='top'></td></tr>
<tr><td valign='top'><span class="indent">Green:</span></td>    <td valign='top'>set highlight color green</td> <td valign='top'></td></tr>
<tr><td valign='top'><span class="indent">Red:</span></td>        <td valign='top'>set highlight color red</td> <td valign='top'></td></tr>

<tr><td colspan="3"><br>n) Highlight Style</td> </tr>
<tr><td valign='top'><span class="indent">Outline:</span></td>              <td valign='top'>set highlight style outline</td> <td valign='top'>me.icn3d.bHighlight = 1;</td></tr>
<tr><td valign='top'><span class="indent">3D Objects:</span></td>         <td valign='top'>set highlight style 3d</td> <td valign='top'>me.icn3d.bHighlight = 2;</td></tr>

<tr><td valign='top'>o) Toggle Highlight</td><td valign='top'>toggle highlight</td> <td valign='top'><a href="#toggleHighlight">toggleHighlight</a></td></tr>

<tr><td colspan="3"><br>p) Switch Highlight Level</td> </tr>
<tr><td valign='top'><span class="indent">&uarr;:</span></td>              <td valign='top'>highlight level up</td> <td valign='top'><a href="#switchHighlightLevelUp">switchHighlightLevelUp<a></td></tr>
<tr><td valign='top'><span class="indent">&darr;:</span></td>         <td valign='top'>highlight level down</td> <td valign='top'><a href="#switchHighlightLevelDown">switchHighlightLevelDown</a></td></tr>

</tbody></table>
<br>
<a name="selectb"></a><b>2b. Selection Definition</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
<div>
In the dialog Select -&gt; Advanced, users can use simple specification to select atoms: <br/><br/>
<b>Specification:</b> In the selection "$1HHO,4N7N.A,B,C:5-10,KRDE,chemicals@CA,C":
<ul>
<li>"$1HHO,4N7N" uses "$" to indicate structure selection.<br/>
<li>".A,B,C" uses "." to indicate chain selection.<br/>
<li>":5-10,KRDE,chemicals" uses the colon ":" to indicate residue selection. Residue selection could be residue number (5-10), one-letter  IUPAC abbreviations (KRDE), or predefined names: "proteins", "nucleotides", "chemicals", "ions", and "water". IUPAC abbreviations can be written either as a contiguous string (e.g., ":KRDE"), in order to find all instances of that sequence in the structure, or they can be separated by commas (e.g., ":K,R,D,E") to select all residues of a given type in the structure (in the latter case, select all Lysine, Arginine, Aspartic Acid, and Glutamic Acid in the structure). <br/>
<li>"@CA,C" uses "@" to indicate atom selection.<br/>
<li>Partial definition is allowed, e.g., ":1-10" selects all residue IDs 1-10 in all chains.<br/>
<li>Different selections can be united (with "<b>or</b>", default), intersected (with "<b>and</b>"), or negated (with "<b>not</b>"). For example, ":1-10 or :K" selects all residues 1-10 and all Lys residues. ":1-10 and :K" selects all Lys residues in the range of residue number 1-10. ":1-10 or not :K" selects all residues 1-10, which are not Lys residues.<br/>
<li>The wild card character \"X\" or \"x\" can be used to represent any character.
</ul>
<b>Set Operation:</b>
<ul>
<li>Users can select multiple sets in the menu "Select > Defined Sets".<br/>
<li>Different sets can be unioned (with "<b>or</b>", default), intersected (with "<b>and</b>"), or excluded (with "<b>not</b>"). For example, if the "Defined Sets" menu has four sets ":1-10", ":11-20", ":5-15", and ":7-8", the command "saved atoms :1-10 or :11-20 and :5-15 not :7-8" unions all residues 1-10 and 11-20 to get the residues 1-20, then intersects with the residues 5-15 to get the residues 5-15, then exclude the residues 7-8 to get the final residues 5-6 and 9-15.
</ul>
<b>Full commands in url or command window:</b>
<ul>
<li>Select without saving the set: select $1HHO,4N7N.A,B,C:5-10,KRDE,chemicals@CA,C<br/>
<li>Select and save: select $1HHO,4N7N.A,B,C:5-10,KRDE,chemicals@CA,C | name my_name
</ul>
</div>
<br>
<a name="view"></a><b>3. View</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
<table width='100%'><tr><td valign='top' width='30%'><b>Menu</b></td><td valign='top' width='40%'><b>Command</b></td><td valign='top'><b>Method</b></td></tr>
<tbody>

<tr><td valign='top'>a) View -&gt; View Only Selection:</td>  <td valign='top'>show selection</td> <td valign='top'><a href="#showSelection">showSelection</a></td></tr>
<tr><td valign='top'>b) View -&gt; Zoom in Selection:</td>  <td valign='top'>zoom selection</td> <td valign='top'><a href="#zoominSelection">zoominSelection</a></td></tr>
<tr><td valign='top'>c) View -&gt; Center Selection:</td>  <td valign='top'>center selection</td> <td valign='top'><a href="#centerSelection">centerSelection</a></td></tr>
<tr><td valign='top'>d) View -&gt; View Full Structure:</td>  <td valign='top'>show all</td> <td valign='top'><a href="#selectAll">showAll</a></td></tr>

<tr><td colspan="3"><br>e) View -&gt; Chemical Binding</td> </tr>
<tr><td valign='top'><span class="indent">Show:</span></td>            <td valign='top'>set chemicalbinding show</td> <td valign='top'>me.setOption('chemicalbinding', 'show');</td></tr>
<tr><td valign='top'><span class="indent">Hide:</span></td>            <td valign='top'>set hbonds off</td> <td valign='top'>me.setOption('chemicalbinding', 'hide');</td></tr>

<tr><td valign='top'>f) View -&gt; H-Bonds & Interactions:</td>  <td valign='top'>hbonds 3.5 | sets non-selected selected;<br>salt bridges 4 | sets non-selected selected;<br>interactions 4 | sets non-selected selected</td> <td valign='top'><a href="#showHbonds">showHbonds;showSaltbridge;pickCustomSphere</a></td></tr>

<tr><td colspan="3"><br>g) View -&gt; Disulfide Bonds</td> </tr>
<tr><td valign='top'><span class="indent">Show:</span></td>            <td valign='top'>disulfide bonds</td> <td valign='top'><a href="#showSsbonds">showSsbonds</a></td></tr>
<tr><td valign='top'><span class="indent">Export Pairs:</span></td>            <td valign='top'>[no command available]</td> <td valign='top'></td></tr>
<tr><td valign='top'><span class="indent">Hide:</span></td>            <td valign='top'>set disulfide bonds off</td> <td valign='top'>me.icn3d.opts["ssbonds"] = "no";</td></tr>

<tr><td colspan="3"><br>h) View -&gt; Cross-Linkages</td> </tr>
<tr><td valign='top'><span class="indent">Show:</span></td>            <td valign='top'>cross linkage</td> <td valign='top'>me.icn3d.bShowCrossResidueBond = true;</td></tr>
<tr><td valign='top'><span class="indent">Hide:</span></td>            <td valign='top'>set cross linkage off</td> <td valign='top'>me.icn3d.bShowCrossResidueBond = false;</td></tr>

<tr><td colspan="3"><br>i) View -&gt; Assembly (PDB, mmCIF, or MMTF files only)</td> </tr>
<tr><td valign='top'><span class="indent">yes:</span></td>                <td valign='top'>set assembly on</td> <td valign='top'>me.icn3d.bAssembly = true;</td></tr>
<tr><td valign='top'><span class="indent">no:</span></td>                 <td valign='top'>set assembly off</td> <td valign='top'>me.icn3d.bAssembly = false;</td></tr>

<tr><td valign='top'>j) View -&gt; Symmetry:</td>  <td valign='top'>symmetry C6 (global)</td> <td valign='top'>retrieveSymmetry; applySymmetry</td></tr>

<tr><td colspan="3"><br>k) View -&gt; Distance</td> </tr>
<tr><td valign='top' rowspan="2"><span class="indent">Measure:</span></td>            <td valign='top'>add label [label text] | x [x] y [y] z [z] | size [label size] | color [FF0000] | background [FFFFFF]</td> <td valign='top'><a href="#addLabel">addLabel</a></td></tr>
<tr>                                                               <td valign='top'>add line | x1 [x1] y1 [y1] z1 [z1] | x2 [x2] y2 [y2] z2 [z2] | color [FFFF00] | dashed [true]</td> <td valign='top'><a href="#addLine">addLine</a></td></tr>
<tr><td valign='top'><span class="indent">Hide:</span></td>            <td valign='top'>set lines off</td> <td valign='top'>me.icn3d.labels['distance'] = []; me.icn3d.lines['distance'] = [];</td></tr>

<tr><td colspan="3"><br>l) View -&gt; Label</td> </tr>
<tr><td valign='top'><span class="indent">by Picking Atoms:</span></td>            <td valign='top'>add label [label text] | x [x] y [y] z [z] | size [label size] | color [FF0000] | background [FFFFFF]</td> <td valign='top'><a href="#addLabel">addLabel</a></td></tr>
<tr><td valign='top'><span class="indent">per Selection:</span></td>            <td valign='top'>add label [label text] | size [label size] | color [FF0000] | background [FFFFFF]</td> <td valign='top'><a href="#addLabel">addLabel</a></td></tr>
<tr><td valign='top'><span class="indent">per Atom:</span></td>            <td valign='top'>add atom labels</td> <td valign='top'><a href="#addAtomLabels">addAtomLabels</a></td></tr>
<tr><td valign='top'><span class="indent">per Residue:</span></td>            <td valign='top'>add residue labels</td> <td valign='top'><a href="#addResiudeLabels">addResiudeLabels</a></td></tr>
<tr><td valign='top'><span class="indent">per Chain:</span></td>            <td valign='top'>add chain labels</td> <td valign='top'><a href="#addLabel">addLabel</a></td></tr>
<tr><td valign='top'><span class="indent">N- & C- Termini:</span></td>            <td valign='top'>add terminal labels</td> <td valign='top'><a href="#addTerminiLabels">addTerminiLabels</a></td></tr>
<tr><td valign='top'><span class="indent">Remove:</span></td>            <td valign='top'>set labels off</td> <td valign='top'><a href="#addChainLabels">addChainLabels</a></td></tr>

<tr><td colspan="3"><br>m) View -&gt; Label Scale</td> </tr>
<tr><td valign='top'><span class="indent">0.1:</span></td>                <td valign='top'>set label scale 0.1</td> <td valign='top'>me.icn3d.labelScale=0.1</td></tr>
<tr><td valign='top'><span class="indent">0.2:</span></td>                <td valign='top'>set label scale 0.2</td> <td valign='top'>me.icn3d.labelScale=0.2</td></tr>
<tr><td valign='top'><span class="indent">0.4:</span></td>                <td valign='top'>set label scale 0.4</td> <td valign='top'>me.icn3d.labelScale=0.4</td></tr>
<tr><td valign='top'><span class="indent">0.6:</span></td>                <td valign='top'>set label scale 0.6</td> <td valign='top'>me.icn3d.labelScale=0.6</td></tr>
<tr><td valign='top'><span class="indent">0.8:</span></td>                <td valign='top'>set label scale 0.8</td> <td valign='top'>me.icn3d.labelScale=0.8</td></tr>
<tr><td valign='top'><span class="indent">1.0:</span></td>                <td valign='top'>set label scale 1.0</td> <td valign='top'>me.icn3d.labelScale=1.0</td></tr>
<tr><td valign='top'><span class="indent">2.0:</span></td>                <td valign='top'>set label scale 2.0</td> <td valign='top'>me.icn3d.labelScale=2.0</td></tr>
<tr><td valign='top'><span class="indent">4.0:</span></td>                <td valign='top'>set label scale 4.0</td> <td valign='top'>me.icn3d.labelScale=4.0</td></tr>
<tr><td valign='top'><span class="indent">6.0:</span></td>                <td valign='top'>set label scale 6.0</td> <td valign='top'>me.icn3d.labelScale=6.0</td></tr>
<tr><td valign='top'><span class="indent">8.0:</span></td>                <td valign='top'>set label scale 8.0</td> <td valign='top'>me.icn3d.labelScale=8.0</td></tr>
<tr><td valign='top'><span class="indent">10.0:</span></td>                <td valign='top'>set label scale 10.0</td> <td valign='top'>me.icn3d.labelScale=10.0</td></tr>

<tr><td valign='top'>n) View -&gt; Show Side by Side:</td>     <td valign='top'>side by side | [url]</td> <td valign='top'>window.open()</td></tr>

<tr><td colspan="3"><br>o) View -&gt; Rotate</td> </tr>

<tr><td colspan="3"><br><span class="indent">m1) Rotate 90&deg;</span></td> </tr>
<tr><td valign='top'><span class="indent">X-axis:</span></td>              <td valign='top'>rotate x</td> <td valign='top'><a href="#rotStruc">setRotation</a></td></tr>
<tr><td valign='top'><span class="indent">Y-axis:</span></td>              <td valign='top'>rotate y</td> <td valign='top'><a href="#rotStruc">setRotation</a></td></tr>
<tr><td valign='top'><span class="indent">Z-axis:</span></td>              <td valign='top'>rotate z</td> <td valign='top'><a href="#rotStruc">setRotation</a></td></tr>

<tr><td colspan="3"><br><span class="indent">m2) Auto Rotation</span></td> </tr>
<tr><td valign='top'><span class="indent">Rotate Left:</span></td>              <td valign='top'>rotate left</td> <td valign='top'><a href="#rotStruc">rotStruc</a></td></tr>
<tr><td valign='top'><span class="indent">Rotate Right:</span></td>              <td valign='top'>rotate right</td> <td valign='top'><a href="#rotStruc">rotStruc</a></td></tr>
<tr><td valign='top'><span class="indent">Rotate Up:</span></td>              <td valign='top'>rotate up</td> <td valign='top'><a href="#rotStruc">rotStruc</a></td></tr>
<tr><td valign='top'><span class="indent">Rotate Down:</span></td>              <td valign='top'>rotate down</td> <td valign='top'><a href="#rotStruc">rotStruc</a></td></tr>

<tr><td colspan="3"><br>p) View -&gt; Camera</td> </tr>
<tr><td valign='top'><span class="indent">Perspective:</span></td>        <td valign='top'>set camera perspective</td> <td valign='top'>me.icn3d.opts['camera'] = 'perspective';</td></tr>
<tr><td valign='top'><span class="indent">Orthographic:</span></td>       <td valign='top'>set camera orthographic</td> <td valign='top'>me.icn3d.opts['camera'] = 'orthographic';</td></tr>

<tr><td colspan="3"><br>q) View -&gt; Fog for Selection</td> </tr>
<tr><td valign='top'><span class="indent">On:</span></td>                <td valign='top'>set fog on</td> <td valign='top'>me.icn3d.opts['fog'] = 'yes';</td></tr>
<tr><td valign='top'><span class="indent">Off:</span></td>                 <td valign='top'>set fog off</td> <td valign='top'>me.icn3d.opts['fog'] = 'no';</td></tr>

<tr><td colspan="3"><br>r) View -&gt; Slab for Selection</td> </tr>
<tr><td valign='top'><span class="indent">On:</span></td>                <td valign='top'>set slab on</td> <td valign='top'>me.icn3d.opts['slab'] = 'yes';</td></tr>
<tr><td valign='top'><span class="indent">Off:</span></td>                 <td valign='top'>set slab off</td> <td valign='top'>me.icn3d.opts['slab'] = 'no';</td></tr>

<tr><td colspan="3"><br>s) View -&gt; XYZ-axes</td> </tr>
<tr><td valign='top'><span class="indent">Show:</span></td>                <td valign='top'>set axis on</td> <td valign='top'>me.icn3d.opts['axis'] = 'yes';</td></tr>
<tr><td valign='top'><span class="indent">Hide:</span></td>                 <td valign='top'>set axis off</td> <td valign='top'>me.icn3d.opts['axis'] = 'no';</td></tr>

<tr><td colspan="3"><br>t) View -&gt; Reset:</td> </tr>
<tr><td valign='top'><span class="indent">All:</span></td>                <td valign='top'>reset</td> <td valign='top'><a href="#renderFinalStep">renderFinalStep</a></td></tr>
<tr><td valign='top'><span class="indent">Orientation:</span></td>                 <td valign='top'>reset orientation</td> <td valign='top'><a href="#resetOrientation">resetOrientation</a></td></tr>

<tr><td valign='top'>u) View -&gt; Undo:</td>     <td valign='top'>back</td> <td valign='top'><a href="#back">back</a></td></tr>

<tr><td valign='top'>v) View -&gt; Redo:</td>     <td valign='top'>forward</td> <td valign='top'><a href="#forward">forward</a></td></tr>

<tr><td valign='top'>w) View -&gt; Full Screen:</td>     <td valign='top'>[no command available]</td> <td valign='top'></td></tr>

</tbody></table>

<br>
<a name="style"></a><b>4. Style</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
<table width='100%'><tr><td valign='top' width='30%'><b>Menu</b></td><td valign='top' width='40%'><b>Command</b></td><td valign='top'><b>Method</b></td></tr>
<tbody>
<tr><td colspan="3">a) Style -&gt; Protein</td> </tr>
<tr><td width="240"><span class="indent">Ribbon:</span></td>             <td valign='top'>style proteins ribbon</td> <td valign='top'><a href="#setStyle">setStyle</a></td></tr>
<tr><td valign='top'><span class="indent">Strand:</span></td>             <td valign='top'>style proteins strand</td> <td valign='top'><a href="#setStyle">setStyle</a></td></tr>
<tr><td valign='top'><span class="indent">Cylinder and Plate:</span></td>   <td valign='top'>style proteins cylinder and plate</td> <td valign='top'><a href="#setStyle">setStyle</a></td></tr>
<tr><td valign='top'><span class="indent">Schematic:</span></td>           <td valign='top'>style proteins schematic</td> <td valign='top'><a href="#setStyle">setStyle</a></td></tr>
<tr><td valign='top'><span class="indent">C Alpha Trace:</span></td>      <td valign='top'>style proteins c alpha trace</td> <td valign='top'><a href="#setStyle">setStyle</a></td></tr>
<tr><td valign='top'><span class="indent">Backbone:</span></td>           <td valign='top'>style proteins backbone</td> <td valign='top'><a href="#setStyle">setStyle</a></td></tr>
<tr><td valign='top'><span class="indent">B-factor Tube:</span></td>      <td valign='top'>style proteins b factor tube</td> <td valign='top'><a href="#setStyle">setStyle</a></td></tr>
<tr><td valign='top'><span class="indent">Lines:</span></td>              <td valign='top'>style proteins lines</td> <td valign='top'><a href="#setStyle">setStyle</a></td></tr>
<tr><td valign='top'><span class="indent">Stick:</span></td>              <td valign='top'>style proteins stick</td> <td valign='top'><a href="#setStyle">setStyle</a></td></tr>
<tr><td valign='top'><span class="indent">Ball and Stick:</span></td>       <td valign='top'>style proteins ball and stick</td> <td valign='top'><a href="#setStyle">setStyle</a></td></tr>
<tr><td valign='top'><span class="indent">Sphere:</span></td>             <td valign='top'>style proteins sphere</td> <td valign='top'><a href="#setStyle">setStyle</a></td></tr>
<tr><td valign='top'><span class="indent">Hide:</span></td>            <td valign='top'>style proteins nothing</td> <td valign='top'><a href="#setStyle">setStyle</a></td></tr>

<tr><td colspan="3"><br>b) Style -&gt; Side Chains</td> </tr>
<tr><td valign='top'><span class="indent">Lines:</span></td>              <td valign='top'>style sidec lines</td> <td valign='top'><a href="#setStyle">setStyle</a></td></tr>
<tr><td valign='top'><span class="indent">Stick:</span></td>              <td valign='top'>style sidec stick</td> <td valign='top'><a href="#setStyle">setStyle</a></td></tr>
<tr><td valign='top'><span class="indent">Ball and Stick:</span></td>       <td valign='top'>style sidec ball and stick</td> <td valign='top'><a href="#setStyle">setStyle</a></td></tr>
<tr><td valign='top'><span class="indent">Sphere:</span></td>             <td valign='top'>style sidec sphere</td> <td valign='top'><a href="#setStyle">setStyle</a></td></tr>
<tr><td valign='top'><span class="indent">Hide:</span></td>            <td valign='top'>style sidec nothing</td> <td valign='top'><a href="#setStyle">setStyle</a></td></tr>

<tr><td colspan="3"><br>c) Style -&gt; Nucleotides</td> </tr>
<tr><td valign='top'><span class="indent">Nucleotide Cartoon:</span></td> <td valign='top'>style nucleotides nucleotide cartoon</td> <td valign='top'><a href="#setStyle">setStyle</a></td></tr>
<tr><td valign='top'><span class="indent">O3' Trace:</span></td>   <td valign='top'>style nucleotides o3 trace</td> <td valign='top'><a href="#setStyle">setStyle</a></td></tr>
<tr><td valign='top'><span class="indent">Backbone:</span></td>            <td valign='top'>style nucleotides backbone</td> <td valign='top'><a href="#setStyle">setStyle</a></td></tr>
<tr><td valign='top'><span class="indent">Schematic:</span></td>            <td valign='top'>style nucleotides schematic</td> <td valign='top'><a href="#setStyle">setStyle</a></td></tr>
<tr><td valign='top'><span class="indent">Lines:</span></td>              <td valign='top'>style nucleotides lines</td> <td valign='top'><a href="#setStyle">setStyle</a></td></tr>
<tr><td valign='top'><span class="indent">Stick:</span></td>              <td valign='top'>style nucleotides stick</td> <td valign='top'><a href="#setStyle">setStyle</a></td></tr>
<tr><td valign='top'><span class="indent">Ball and Stick:</span></td>       <td valign='top'>style nucleotides ball and stick</td> <td valign='top'><a href="#setStyle">setStyle</a></td></tr>
<tr><td valign='top'><span class="indent">Sphere:</span></td>             <td valign='top'>style nucleotides sphere</td> <td valign='top'><a href="#setStyle">setStyle</a></td></tr>
<tr><td valign='top'><span class="indent">Hide:</span></td>            <td valign='top'>style nucleotides nothing</td> <td valign='top'><a href="#setStyle">setStyle</a></td></tr>

<tr><td colspan="3"><br>d) Style -&gt; Chemicals</td> </tr>
<tr><td valign='top'><span class="indent">Lines:</span></td>              <td valign='top'>style chemicals lines</td> <td valign='top'><a href="#setStyle">setStyle</a></td></tr>
<tr><td valign='top'><span class="indent">Stick:</span></td>              <td valign='top'>style chemicals stick</td> <td valign='top'><a href="#setStyle">setStyle</a></td></tr>
<tr><td valign='top'><span class="indent">Ball and Stick:</span></td>       <td valign='top'>style chemicals ball and stick</td> <td valign='top'><a href="#setStyle">setStyle</a></td></tr>
<tr><td valign='top'><span class="indent">Schematic:</span></td>           <td valign='top'>style chemicals schematic</td> <td valign='top'><a href="#setStyle">setStyle</a></td></tr>
<tr><td valign='top'><span class="indent">Sphere:</span></td>             <td valign='top'>style chemicals sphere</td> <td valign='top'><a href="#setStyle">setStyle</a></td></tr>
<tr><td valign='top'><span class="indent">Hide:</span></td>            <td valign='top'>style chemicals nothing</td> <td valign='top'><a href="#setStyle">setStyle</a></td></tr>

<tr><td colspan="3"><br>d2) Style -&gt; Hydrogens</td> </tr>
<tr><td valign='top'><span class="indent">Show:</span></td>              <td valign='top'>hydrogens</td> <td valign='top'><a href="#setStyle">showHydrogens</a></td></tr>
<tr><td valign='top'><span class="indent">Hide:</span></td>            <td valign='top'>set hydrogens off</td> <td valign='top'><a href="#setStyle">hideHydrogens</a></td></tr>

<tr><td colspan="3"><br>e) Style -&gt; Ions</td> </tr>
<tr><td valign='top'><span class="indent">Sphere:</span></td>             <td valign='top'>style ions sphere</td> <td valign='top'><a href="#setStyle">setStyle</a></td></tr>
<tr><td valign='top'><span class="indent">Dot:</span></td>                <td valign='top'>style ions dot</td> <td valign='top'><a href="#setStyle">setStyle</a></td></tr>
<tr><td valign='top'><span class="indent">Hide:</span></td>            <td valign='top'>style ions nothing</td> <td valign='top'><a href="#setStyle">setStyle</a></td></tr>

<tr><td colspan="3"><br>f) Style -&gt; Water</td> </tr>
<tr><td valign='top'><span class="indent">Sphere:</span></td>             <td valign='top'>style water sphere</td> <td valign='top'><a href="#setStyle">setStyle</a></td></tr>
<tr><td valign='top'><span class="indent">Dot:</span></td>                <td valign='top'>style water dot</td> <td valign='top'><a href="#setStyle">setStyle</a></td></tr>
<tr><td valign='top'><span class="indent">Hide:</span></td>            <td valign='top'>style water nothing</td> <td valign='top'><a href="#setStyle">setStyle</a></td></tr>

<tr><td valign='top'>g) Style -&gt; Set Thickness:</td>                    <td valign='top'>set thickness | linerad 0.6 | coilrad 1 | stickrad 0.8 | ribbonthick 0.8 | proteinwidth 2 | nucleotidewidth 1.2 | ballscale 0.6</td> <td valign='top'></td></tr>

<tr><td valign='top'>h) Style -&gt; Save Style:</td>                    <td valign='top'>save style</td> <td valign='top'><a href="#saveStyle">saveStyle</a></td></tr>
<tr><td valign='top'>i) Style -&gt; Apply Saved Style:</td>             <td valign='top'>apply saved style</td> <td valign='top'><a href="#applySavedStyle">applySavedStyle</a></td></tr>

<tr><td colspan="3"><br>j) Style -&gt; Surface Type</td> </tr>
<tr><td width="240"><span class="indent">Van der Waals:</span></td>      <td valign='top'>set surface Van der Waals surface</td> <td valign='top'><a href="#applySurfaceOptions">applySurfaceOptions</a></td></tr>
<tr><td width="240"><span class="indent">VDW with Context:</span></td>      <td valign='top'>set surface Van der Waals surface with context</td> <td valign='top'><a href="#applySurfaceOptions">applySurfaceOptions</a></td></tr>
<tr><td valign='top'><span class="indent">Molecular Surface:</span></td>          <td valign='top'>set surface molecular surface</td> <td valign='top'><a href="#applySurfaceOptions">applySurfaceOptions</a></td></tr>
<tr><td valign='top'><span class="indent">MS with Context:</span></td>          <td valign='top'>set surface molecular surface with context</td> <td valign='top'><a href="#applySurfaceOptions">applySurfaceOptions</a></td></tr>
<tr><td valign='top'><span class="indent">Solvent Accessible:</span></td> <td valign='top'>set surface solvent accessible surface</td> <td valign='top'><a href="#applySurfaceOptions">applySurfaceOptions</a></td></tr>
<tr><td valign='top'><span class="indent">SA with Context:</span></td> <td valign='top'>set surface solvent accessible surface with context</td> <td valign='top'><a href="#applySurfaceOptions">applySurfaceOptions</a></td></tr>

<tr><td valign='top'>k) Style -&gt; Remove Surface:</td>                    <td valign='top'>set surface nothing</td> <td valign='top'><a href="#applySurfaceOptions">applySurfaceOptions</a></td></tr>

<tr><td colspan="3"><br>l) Style -&gt; Surface Opacity</td> </tr>
<tr><td valign='top'><span class="indent">1.0:</span></td>                <td valign='top'>set surface opacity 1.0</td> <td valign='top'><a href="#applySurfaceOptions">applySurfaceOptions</a></td></tr>
<tr><td valign='top'><span class="indent">0.9:</span></td>                <td valign='top'>set surface opacity 0.9</td> <td valign='top'><a href="#applySurfaceOptions">applySurfaceOptions</a></td></tr>
<tr><td valign='top'><span class="indent">0.8:</span></td>                <td valign='top'>set surface opacity 0.8</td> <td valign='top'><a href="#applySurfaceOptions">applySurfaceOptions</a></td></tr>
<tr><td valign='top'><span class="indent">0.7:</span></td>                <td valign='top'>set surface opacity 0.7</td> <td valign='top'><a href="#applySurfaceOptions">applySurfaceOptions</a></td></tr>
<tr><td valign='top'><span class="indent">0.6:</span></td>                <td valign='top'>set surface opacity 0.6</td> <td valign='top'><a href="#applySurfaceOptions">applySurfaceOptions</a></td></tr>
<tr><td valign='top'><span class="indent">0.5:</span></td>                <td valign='top'>set surface opacity 0.5</td> <td valign='top'><a href="#applySurfaceOptions">applySurfaceOptions</a></td></tr>
<tr><td valign='top'><span class="indent">0.4:</span></td>                <td valign='top'>set surface opacity 0.4</td> <td valign='top'><a href="#applySurfaceOptions">applySurfaceOptions</a></td></tr>
<tr><td valign='top'><span class="indent">0.3:</span></td>                <td valign='top'>set surface opacity 0.3</td> <td valign='top'><a href="#applySurfaceOptions">applySurfaceOptions</a></td></tr>
<tr><td valign='top'><span class="indent">0.2:</span></td>                <td valign='top'>set surface opacity 0.2</td> <td valign='top'><a href="#applySurfaceOptions">applySurfaceOptions</a></td></tr>
<tr><td valign='top'><span class="indent">0.1:</span></td>                <td valign='top'>set surface opacity 0.1</td> <td valign='top'><a href="#applySurfaceOptions">applySurfaceOptions</a></td></tr>

<tr><td colspan="3"><br>m) Style -&gt; Surface Wireframe</td> </tr>
<tr><td valign='top'><span class="indent">Yes</span></td>                 <td valign='top'>set surface wireframe on</td> <td valign='top'><a href="#applySurfaceOptions">applySurfaceOptions</a></td></tr>
<tr><td valign='top'><span class="indent">No</span></td>                  <td valign='top'>set surface wireframe off</td> <td valign='top'><a href="#applySurfaceOptions">applySurfaceOptions</a></td></tr>

<tr><td colspan="3"><br>n) Style -&gt; Electron Density</td> </tr>
<tr><td width="240"><span class="indent">2Fo-Fc Map:</span></td>      <td valign='top'>set map 2fofc sigma 1.5</td> <td valign='top'><a href="#applyCommandMap">applyCommandMap</a></td></tr>
<tr><td width="240"><span class="indent">Fo-Fc Map:</span></td>      <td valign='top'>set map 2fofc sigma 3</td> <td valign='top'><a href="#applyCommandMap">applyCommandMap</a></td></tr>

<tr><td valign='top'>o) Style -&gt; Remove Map:</td>                    <td valign='top'>set map nothing</td> <td valign='top'><a href="#applyMapOptions">applyMapOptions</a></td></tr>

<tr><td colspan="3"><br>p) Style -&gt; Map Wireframe</td> </tr>
<tr><td valign='top'><span class="indent">Yes</span></td>                 <td valign='top'>set map wireframe on</td> <td valign='top'><a href="#applyMapOptions">applyMapOptions</a></td></tr>
<tr><td valign='top'><span class="indent">No</span></td>                  <td valign='top'>set map wireframe off</td> <td valign='top'><a href="#applyMapOptions">applyMapOptions</a></td></tr>

<tr><td valign='top'>q) Style -&gt; EM Density Map:</td>                    <td valign='top'>set emmap percentage 20</td> <td valign='top'><a href="#applyCommandEmmap">applyCommandEmmap</a></td></tr>

<tr><td valign='top'>r) Style -&gt; Remove EM Map:</td>                    <td valign='top'>set emmap nothing</td> <td valign='top'><a href="#applyEmmapOptions">applyEmmapOptions</a></td></tr>

<tr><td colspan="3"><br>s) Style -&gt; EM Map Wireframe</td> </tr>
<tr><td valign='top'><span class="indent">Yes</span></td>                 <td valign='top'>set emmap wireframe on</td> <td valign='top'><a href="#applyEmmapOptions">applyEmmapOptions</a></td></tr>
<tr><td valign='top'><span class="indent">No</span></td>                  <td valign='top'>set emmap wireframe off</td> <td valign='top'><a href="#applyEmmapOptions">applyEmmapOptions</a></td></tr>

<tr><td colspan="3"><br>t) Style -&gt; Background</td> </tr>
<tr><td valign='top'><span class="indent">Transparent:</span></td>        <td valign='top'>set background transparent</td> <td valign='top'>me.icn3d.opts['background'] = 'transparent'</td></tr>
<tr><td valign='top'><span class="indent">Black:</span></td>              <td valign='top'>set background black</td> <td valign='top'>me.icn3d.opts['background'] = 'black'</td></tr>
<tr><td valign='top'><span class="indent">Grey:</span></td>               <td valign='top'>set background grey</td> <td valign='top'>me.icn3d.opts['background'] = 'grey'</td></tr>
<tr><td valign='top'><span class="indent">White:</span></td>              <td valign='top'>set background white</td> <td valign='top'>me.icn3d.opts['background'] = 'white'</td></tr>

</tbody></table>
<br>
<a name="color"></a><b>5. Color</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
<table width='100%'><tr><td valign='top' width='30%'><b>Menu</b></td><td valign='top' width='40%'><b>Command</b></td><td valign='top'><b>Method</b></td></tr>
<tbody><tr><td colspan="3">a)</td> </tr>
<tr><td width="240">Color -&gt; Spectrum:</td>          <td valign='top'>color spectrum</td> <td valign='top'><a href="#setColorByOptions">setColorByOptions</a></td></tr>
<tr><td width="240">Color -&gt; Secondary (Sheet in Green):</td>         <td valign='top'>color secondary structure</td> <td valign='top'><a href="#setColorByOptions">setColorByOptions</a></td></tr>
<tr><td width="240">Color -&gt; Secondary (Sheet in Yellow):</td>        <td valign='top'>color secondary structure yellow</td> <td valign='top'><a href="#setColorByOptions">setColorByOptions</a></td></tr>
<tr><td width="240">Color -&gt; Secondary (Spectrum):</td>        <td valign='top'>color secondary structure spectrum</td> <td valign='top'><a href="#setColorByOptions">setColorByOptions</a></td></tr>
<tr><td valign='top'>Color -&gt; Charge:</td>           <td valign='top'>color charge</td> <td valign='top'><a href="#setColorByOptions">setColorByOptions</a></td></tr>
<tr><td valign='top'>Color -&gt; Wimley-White Hydrophobicity:</td>      <td valign='top'>color hydrophobic</td> <td valign='top'><a href="#setColorByOptions">setColorByOptions</a></td></tr>
<tr><td valign='top'>Color -&gt; B-factor (Original):</td>             <td valign='top'>color b factor</td> <td valign='top'><a href="#setColorByOptions">setColorByOptions</a></td></tr>
<tr><td valign='top'>Color -&gt; B-factor (Percentile):</td>             <td valign='top'>color b factor percentile</td> <td valign='top'><a href="#setColorByOptions">setColorByOptions</a></td></tr>
<tr><td valign='top'>Color -&gt; Solvent Accessibility:</td>             <td valign='top'>color area | 35</td> <td valign='top'><a href="#setColorByOptions">setColorByOptions</a></td></tr>
<tr><td valign='top'>Color -&gt; Chain:</td>            <td valign='top'>color chain</td> <td valign='top'><a href="#setColorByOptions">setColorByOptions</a></td></tr>
<tr><td valign='top'>Color -&gt; 3D Domain:</td>          <td valign='top'>color 3ddomain</td> <td valign='top'><a href="#setColorByOptions">setColorByOptions</a></td></tr>
<tr><td valign='top'>Color -&gt; Residue - Default:</td>          <td valign='top'>color residue</td> <td valign='top'><a href="#setColorByOptions">setColorByOptions</a></td></tr>
<tr><td valign='top'>Color -&gt; Residue - Custom:</td>          <td valign='top'>color residue custom</td> <td valign='top'><a href="#setColorByOptions">setColorByOptions</a></td></tr>
<tr><td valign='top'>Color -&gt; Atom:</td>             <td valign='top'>color atom</td> <td valign='top'><a href="#setColorByOptions">setColorByOptions</a></td></tr>

<tr><td colspan="3"><br>b) Color -&gt; Unicolor</td> </tr>
<tr><td valign='top'><span class="indent">Red:</span></td>                <td valign='top'>color red</td> <td valign='top'><a href="#setColorByOptions">setColorByOptions</a></td></tr>
<tr><td valign='top'><span class="indent">Green:</span></td>              <td valign='top'>color green</td> <td valign='top'><a href="#setColorByOptions">setColorByOptions</a></td></tr>
<tr><td valign='top'><span class="indent">Blue:</span></td>               <td valign='top'>color blue</td> <td valign='top'><a href="#setColorByOptions">setColorByOptions</a></td></tr>
<tr><td valign='top'><span class="indent">Magenta:</span></td>            <td valign='top'>color magenta</td> <td valign='top'><a href="#setColorByOptions">setColorByOptions</a></td></tr>
<tr><td valign='top'><span class="indent">Yellow:</span></td>             <td valign='top'>color yellow</td> <td valign='top'><a href="#setColorByOptions">setColorByOptions</a></td></tr>
<tr><td valign='top'><span class="indent">Cyan:</span></td>               <td valign='top'>color cyan</td> <td valign='top'><a href="#setColorByOptions">setColorByOptions</a></td></tr>
<tr><td valign='top'><span class="indent">White:</span></td>              <td valign='top'>color white</td> <td valign='top'><a href="#setColorByOptions">setColorByOptions</a></td></tr>
<tr><td valign='top'><span class="indent">Grey:</span></td>               <td valign='top'>color grey</td> <td valign='top'><a href="#setColorByOptions">setColorByOptions</a></td></tr>

<tr><td colspan="3"><br>c) Color -&gt; Color Picker</td> </tr>
<tr><td valign='top'><span class="indent">Apply:</span></td>              <td valign='top'>color [e.g., FF0000]</td> <td valign='top'><a href="#setColorByOptions">setColorByOptions</a></td></tr>

<tr><td valign='top'>Color -&gt; Save Color:</td>       <td valign='top'>save color</td> <td valign='top'><a href="#saveColor">saveColor</a></td></tr>
<tr><td valign='top'>Color -&gt; Apply Saved Color:</td><td valign='top'>apply saved color</td> <td valign='top'><a href="#applySavedColor">applySavedColor</a></td></tr>

</tbody></table>
<br>
<a name="windows"></a><b>6. Windows</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
<table width='100%'><tr><td valign='top' width='30%'><b>Menu</b></td><td valign='top' width='40%'><b>Command</b></td><td valign='top'><b>Method</b></td></tr>
<tbody>
<tr><td width="240">a) Windows -&gt; View Sequences & Annotations:</td>                <td valign='top'>view annotations</td> <td valign='top'><a href="#applyCommandAnnotationsAndCddSite">applyCommandAnnotationsAndCddSite</a></td></tr>
<tr><td valign='top'>b) Windows -&gt; View Interactions</td>         <td valign='top'>view interactions</td> <td valign='top'><a href="#set2DDiagrams">set2DDiagrams</a></td></tr>
<tr><td valign='top'>c) Windows -&gt; Defined Sets</td>     <td valign='top'>defined sets</td> <td valign='top'><a href="#showSets">showSets</a></td></tr>

<tr><td valign='top'>c) Windows -&gt; Your Note / Window Title</td>     <td valign='top'>your note | [text]</td> <td valign='top'></td></tr>

<tr><td colspan="3"><br>d) Windows -&gt; View Links</td> </tr>
<tr><td valign='top'><span class="indent">Structure Summary:</span></td>          <td valign='top'>[no command available]</td> <td valign='top'></td></tr>
<tr><td valign='top'><span class="indent">Find Similar Structures:</span></td>   <td valign='top'>[no command available]</td> <td valign='top'></td></tr>
<tr><td valign='top'><span class="indent">Literature:</span></td>                <td valign='top'>[no command available]</td> <td valign='top'></td></tr>
<tr><td valign='top'><span class="indent">Protein:</span></td>                <td valign='top'>[no command available]</td> <td valign='top'></td></tr>

</tbody></table>
<br>
<a name="help"></a><b>7. Help</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
<table width='100%'><tr><td width='30%'><b>Menu</b></td><td width='40%'><b>Command</b></td> <td></td></tr>
<tr><td valign='top'>a) Help -&gt; About iCn3D:</td>      <td valign='top'>[no command available]</td> <td></td></tr>
<tr><td valign='top'>b) Help -&gt; Help DOc:</td>      <td valign='top'>[no command available]</td> <td></td></tr>
<tr><td valign='top'>c) Help -&gt; Web Methods:</td>      <td valign='top'>[no command available]</td> <td></td></tr>
<tr><td valign='top'>d) Help -&gt; Transform Hints:</td>      <td valign='top'>[no command available]</td> <td></td></tr>
<tr><td valign='top'>e) Help -&gt; Selection Hints:</td>      <td valign='top'>[no command available]</td> <td></td></tr>
</tbody></table>

<hr>

    <a name="DataStructure"></a>
    <h2>Data Structure<img src="https://www.ncbi.nlm.nih.gov/Structure/IMG/spacer.gif" width="25" height="1" border="0"><a href="#Top"><img src="https://www.ncbi.nlm.nih.gov/Structure/IMG/arrowup_blue.gif" width="12" height="12" border="0" alt="back to top"></a></h2>
    <ul>
    <li><b>Atom Object</b>
      <pre>      var atomDetails = {
        serial: serial,         // required, unique atom id
        structure: structure,   // required, used to identify the structure
        chain: chain,           // required, used to identify the chain
        resi: resi,             // required, used to identify residue ID, has to be a integer
        name: atom,             // required, atom name
        coord: coord,           // required, used to draw 3D shape
        coord2: ribbonCoord,    // optional, ribbon's real positions, used to draw stabilizers for 3D printing
        bonds: [],              // required, used to connect atoms

        color: color,           // optional, used to assign atom color, default is grey
        style: style,           // optional, used to assign atom style as one of 13 styles: ribbon, strand, cylinder and plate,
                                //   nucleotide cartoon, o3' trace, schematic, c alpha trace, b factor tube, lines, stick,
                                //   ball and stick, sphere, dot, nothing
        style2: sideChainStyle, // optional, used to assign protein side chain style as one of 13 styles: ribbon, strand, cylinder and plate,
                                //   nucleotide cartoon, o3' trace, schematic, c alpha trace, b factor tube, lines, stick,
                                //   ball and stick, sphere, dot, nothing
        het: false,             // optional, used to determine chemicals, water, ions, etc
        resn: resn,             // optional, used to determine protein or nucleotide
        b: b,                   // optional, used to draw B-factor tube
        elem: elem,             // optional, used to determine hydrogen bond
        ss: 'coil',             // optional, used to show secondary structures
        ssbegin: false,         // optional, used to show the beginning of secondary structures
        ssend: false            // optional, used to show the end of secondary structures
      }
      </pre>
    </li>
    <li><a name="DataInputs"><b>Data required from the input to set up 3D viewer</b></a><br>
    <pre>
        atoms: {};          // REQUIRED, all atoms in the input: atom index => atom details, ONLY THIS HASH STORE ALL ATOM DETAILS
        dAtoms: {};         // REQUIRED, atoms used to display: atom index => 1
        pmin: pmin;         // REQUIRED, the position with minimum x,y,z
        pmax: pmax;         // REQUIRED, the position with maximum x,y,z
        cnt: cnt;           // REQUIRED, total number of atoms
        maxD: pmax.distanceTo(pmin);             // REQUIRED, max dimension of the structure
        center: psum.multiplyScalar(1.0 / cnt);  // REQUIRED, center position of the structure

        hAtoms: {};         // OPTIONAL, atoms used to highlight: atom index => 1
        structures: {};         // OPTIONAL, structure name => array of chain IDs
        chains: {};           // OPTIONAL, structure_chain name => (atom index => 1)
        residues: {};          // OPTIONAL, structure_chain_resi name => (atom index => 1)

        chainsSeq: {};        // OPTIONAL, structure_chain name => array of residue object: {"name":[residue name], "resi": [residue number]}
        chainsAn: {};         // OPTIONAL, structure_chain name => array of annotation
        chainsAnTitle: [];    // OPTIONAL, the titles for the array of annotation
        molTitle: "";       // OPTIONAL, ID and name
        hbondpnts: [];      // OPTIONAL, array of positions of hydrogen bond
        ssbondpnts: {};     // OPTIONAL, structure name => positions of disulfide bonds
        residueId2Name: {}; // OPTIONAL, structure_chain_resi => one letter abbreviation

        proteins: {};       // OPTIONAL, proteins: atom index => 1
        sidec: {};          // OPTIONAL, protein sidechains: atom index => 1
        calphas: {};        // OPTIONAL, protein C alphas: atom index => 1
        nucleotides: {};    // OPTIONAL, DNA, RNA: atom index => 1
        nucleotidesO3: {};  // OPTIONAL, DNA, RNA O3': atom index => 1
        chemicals: {};        // OPTIONAL, chemicals: atom index => 1
        ions: {};           // OPTIONAL, ions: atom index => 1
        water: {};          // OPTIONAL, water: atom index => 1
    </pre>
    </li>
    </ul>

    <hr>

    <a name="DisplayOptions"></a>
    <h2>Display Options<img src="https://www.ncbi.nlm.nih.gov/Structure/IMG/spacer.gif" width="25" height="1" border="0"><a href="#Top"><img src="https://www.ncbi.nlm.nih.gov/Structure/IMG/arrowup_blue.gif" width="12" height="12" border="0" alt="back to top"></a></h2>
    <ul>
    <li><b>Default values and possible choices of display options</b>. (There are two ways to modify the default display. First, you can manually operate in iCn3D viewer to get your custom display. Then click "File > Share Link" to get your desired full URL or short URL. Second, you can update the source code of the file full.html in the lines of "iCn3DUI.prototype.modifyIcn3d".)
      <pre>          var opts = {};
          opts['camera']             = 'perspective';        //perspective, orthographic
          opts['background']         = 'transparent';        //transparent, black, grey, white
          opts['color']              = 'chain';              //spectrum, secondary structure, charge, hydrophobic, chain, residue, atom, b factor
                                                             // red, green, blue, magenta, yellow, cyan, white, grey, custom
          opts['proteins']           = 'ribbon';             //ribbon, strand, cylinder and plate, schematic, c alpha trace,
                                                             // b factor tube, lines, stick, ball and stick, sphere, nothing
          opts['sidec']              = 'nothing';            //lines, stick, ball and stick, sphere, nothing
          opts['nucleotides']        = 'o3 trace';           //nucleotide cartoon, o3 trace, schematic, lines, stick,
                                                             // nucleotides ball and stick, sphere, nothing
          opts['chemicals']          = 'stick';              //line, stick, ball and stick, schematic, sphere, nothing
          opts['water']              = 'nothing';            //sphere, dot, nothing
          opts['ions']               = 'sphere';             //sphere, dot, nothing
          opts['surface']            = 'nothing';            //van der waals surface, van der waals surface with context,
                                                             //molecular surface, molecular surface with context,
                                                             //solvent accessible surface, solvent accessible surface with context,
                                                             //nothing
          opts['opacity']            = '1.0';                //1.0, 0.9, 0.8, 0.7, 0.6, 0.5, 0.4, 0.3, 0.2, 0.1
          opts['wireframe']          = 'no';                 //yes, no
          opts['hbonds']             = 'no';                 //yes, no
          opts['saltbridge']         = 'no';                 //yes, no
          opts['ssbonds']            = 'yes';                //yes, no
          opts['chemicalbinding']    = 'hide';               //show, hide
          opts['axis']               = 'no';                 //yes, no
          opts['fog']                = 'no';                 //yes, no
          opts['slab']               = 'no';                 //yes, no
          opts['pk']                 = 'residue';            //no, atom, residue, strand, chain
      </pre>
    </li>
    </ul>

    <hr>

<!-- ======================= EXAMPLES ======================= -->

    <a name="Methods"></a>
    <h2>Method Documents of the core library iCn3D<img src="https://www.ncbi.nlm.nih.gov/Structure/IMG/spacer.gif" width="25" height="1" border="0"><a href="#Top"><img src="https://www.ncbi.nlm.nih.gov/Structure/IMG/arrowup_blue.gif" width="12" height="12" border="0" alt="back to top"></a></h2>

<h3>Options <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a></h3>
<br>
<a name="bHighlight"></a><b>bHighlight</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
If its value is 1, the selected atoms will be highlighted with outlines around the structure. If its value is 2, the selected atoms will be highlighted with transparent 3D objects such as boxes, ribbons, cylinders, etc. If its value is undefined, no highlight will be shown.<br>
<a name="bImpo"></a><b>bImpo</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
This is a flag to turn on the rendering of spheres and cylinders using shaders instead of geometries. It's true by default if the browser supports the EXT_frag_depth extension.<br>
<br>
<a name="bRender"></a><b>bRender</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
This is a flag to turn off the rendering part if a sequence of commands are executed. It's true by default.<br>
<br>
<a name="bStopRotate"></a><b>bStopRotate</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Once clicked, this flag can be set as "true" to the automatic rotation. It's false by default.<br>
<br>
<a name="coilWidth"></a><b>coilWidth</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
This is the coil radius for coils. It's 0.4 by default.<br>
<br>
<a name="cylinderHelixRadius"></a><b>cylinderHelixRadius</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
This is the cylinder radius for the style 'Cylinder and Plate'. It's 1.6 by default.<br>
<br>
<a name="dAtoms"></a><b>dAtoms</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
This is a hash used to store all atoms to be displayed. The key is the atom index. Its value is set as 1.<br>
<br>
<a name="dotSphereScale"></a><b>dotSphereScale</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
This is the ball scale for styles 'Ball and Stick' and 'Dot'. It's 0.3 by default.<br>
<br>
<a name="hAtoms"></a><b>hAtoms</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
This is a hash used to store all atoms to be highlighted. The key is the atom index. Its value is set as 1.<br>
<br>
<a name="hColor"></a><b>hColor</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
This defines the highlight color.<br>
<br>
<a name="helixSheetWidth"></a><b>helixSheetWidth</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
This is the width of protein ribbons. It's 1.3 by default.<br>
<br>
<a name="lineRadius"></a><b>lineRadius</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
This is the line radius for stabilizers, hydrogen bonds, and distance lines. It's 0.1 by default.<br>
<br>
<a name="matShader"></a><b>matShader</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
This defines the highlight color using the outline method. It can be defined using the function setOutlineColor().<br>
<br>
<a name="nucleicAcidWidth"></a><b>nucleicAcidWidth</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
This is the width of nucleotide ribbons. It's 0.8 by default.<br>
<br>
<a name="pAtom"></a><b>pAtom</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
"pAtom" has the value of the atom index of the picked atom.<br>
<br>
<a name="pAtom2"></a><b>pAtom2</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
When two atoms are required to be selected (e.g., for measuring distance), "pAtom2" has the value of the atom index of the 2nd picked atom.<br>
<br>
<a name="pk"></a><b>pk</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
If its value is 1, selecting an atom will select the atom. If its value is 2, selecting an atom will select the residue containing this atom. If its value is 3, selecting an atom will select the strand or helix or coil containing this atom. If its value is 0, no selecting will work.<br>
<br>
<a name="ribbonthickness"></a><b>ribbonthickness</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
This is the ribbon thickness for helix and sheet ribbons, and nucleotide ribbons. It's 0.4 by default.<br>
<br>
<a name="sphereRadius"></a><b>sphereRadius</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
This is the sphere radius for the style 'Sphere'. It's 1.5 by default.<br>
<br>
<a name="stickRadius"></a><b>stickRadius</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
This is the stick radius for sticks. It's 0.4 by default.<br>
<br>
<a name="traceRadius"></a><b>traceRadius</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
This is the stick radius for C alpha trace and O3' trace. It's 0.2 by default.<br>
<br>

<br>

<h3>Methods <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a></h3>

<br>
<a name="addHlObjects"></a><b>addHlObjects()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Show the highlight for the selected atoms: <a href="#hAtoms">hAtoms</a>.<br>
<br>

<a name="addResiudeLabels"></a><b>addResiudeLabels(atoms, bSchematic)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Add labels for all residues containing the input "atoms". The labels are one-letter residue abbreviations. If "bSchematic" is true, the labels are in circles. Otherwise, they are in round-corner rectangles.<br>
<br>

<a name="applyDisplayOptions"></a><b>applyDisplayOptions(options, atoms)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Apply style and label options to a certain set of atoms.<br>
<br>
<a name="applyOtherOptions"></a><b>applyOtherOptions(options)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Apply the rest options (e.g., hydrogen bonds, center, etc).<br>
<br>
<a name="applySsbondsOptions"></a><b>applySsbondsOptions(options)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Apply the disulfide bond options.<br>
<br>
<a name="applyEmmapOptions"></a><b>applyEmmapOptions(options)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Apply options for EM density map.<br>
<br>
<a name="applyMapOptions"></a><b>applyMapOptions(options)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Apply options for electron density map.<br>
<br>
<a name="applySurfaceOptions"></a><b>applySurfaceOptions(options)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Apply the surface options.<br>
<br>
<a name="applyTransformation"></a><b>applyTransformation()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Update the rotation, translation, and zooming before rendering. Typically used before the function render().<br>
<br>
<a name="buildAxes"></a><b>buildAxes(radius)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Build the xyz-axes from the center of atoms. The maximum axes length is equal to "radius" in angstrom.<br>
<br>
<a name="calculateChemicalHbonds"></a><b>calculateChemicalHbonds(protein, chemicals, threshold)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Set up hydrogen bonds between chemical and protein/nucleotide in the same structure. "protein" and "chemicals" are hashes with atom indices as keys and 1 as values. "threshold" is the maximum distance of hydrogen bonds and has the unit of angstrom.<br>
<br>
<a name="centerAtoms"></a><b>centerAtoms(atoms)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Return an object {"center": center, "maxD": maxD}, where "center" is the center of a set of "atoms" with a value of THREE.Vector3(), and "maxD" is the maximum distance between any two atoms in the set.<br>
<br>
<a name="centerSelection"></a><b>centerSelection()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Center on the selected atoms.<br>
<br>
<a name="createBox"></a><b>createBox(atom, defaultRadius, forceDefault, scale, color, bHighlight)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Create a cube for "atom" with the "defaultRadius". "forceDefault" means to use the default radius. "scale" means scale on the radius. "color" means the color of the cube. "bHighlight" is an option to draw the highlight for the atom.<br>
<br>
<a name="createCylinderCurve"></a><b>createCylinderCurve(atoms, atomNameArray, radius, bLines, bHighlight)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Create small cylinders (thick lines) for "atoms", whose atom name should be in the array atomNameArray. "radius" is radius of the small cylinders. "bLine" is an option to show the cylinders as lines. "bHighlight" is an option to draw the highlight for these atoms. The highlight could be outlines with bHighlight=1 and 3D objects with bHighlight=2 as mentioned <a href="#bHighlight">above</a>.<br>
<br>
<a name="createCylinderHelix"></a><b>createCylinderHelix(atoms, radius, bHighlight)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Create cylinders for alpha helices and ribbons for beta strands in "atoms". "radius" is radius of the cylinders. "bHighlight" is an option to draw the highlight for these atoms. The highlight could be outlines with bHighlight=1 and 3D objects with bHighlight=2 as mentioned <a href="#bHighlight">above</a>.<br>
<br>
<a name="createLabelRepresentation"></a><b>createLabelRepresentation(labels)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Create labels for a list of "labels", each of which has the properties 'position', 'text', 'size', 'color', and 'background'.<br>
<br>
<a name="createLineRepresentation"></a><b>createLineRepresentation(atoms, bHighlight)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Create lines for "atoms". "bHighlight" is an option to draw the highlight for these atoms. The highlight could be outlines with bHighlight=1 and 3D objects with bHighlight=2 as mentioned <a href="#bHighlight">above</a>.<br>
<br>
<a name="createLines"></a><b>createLines(lines)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Create lines for a list of "lines", each of which has the properties 'position1', 'position2', 'color', and a boolean of 'dashed'.<br>
<br>
<a name="createSphereRepresentation"></a><b>createSphereRepresentation(atoms, radius, forceDefault, scale, bHighlight)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Create spheres for "atoms" with the "radius". "forceDefault" means to use the default radius. "scale" means scale on the radius. "bHighlight" is an option to draw the highlight for these atoms. The highlight could be outlines with bHighlight=1 and 3D objects with bHighlight=2 as mentioned <a href="#bHighlight">above</a>.<br>
<br>
<a name="createStickRepresentation"></a><b>createStickRepresentation(atoms, atomR, bondR, scale, bHighlight)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Create sticks for "atoms". "bondR" is the radius of the sticks. "atomR" is the radius of the spheres in the joints. "scale" means scale on the radius. "bHighlight" is an option to draw the highlight for these atoms. The highlight could be outlines with bHighlight=1 and 3D objects with bHighlight=2 as mentioned <a href="#bHighlight">above</a>.<br>
<br>
<a name="createStrand"></a><b>createStrand(atoms, num, div, fill, coilWidth, helixSheetWidth, doNotSmoothen, thickness, bHighlight)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Create the style of ribbon or strand for "atoms". "num" means how many lines define the curve. "num" is 2 for ribbon and 6 for strand. "div" means how many pnts are used to smooth the curve. It's typically 5. "coilWidth" is the width of curve for coil. "helixSheetWidth" is the width of curve for helix or sheet. "doNotSmoothen" is a flag to smooth the curve or not. "thickness" is the thickness of the curve. "bHighlight" is an option to draw the highlight for these atoms. The highlight could be outlines with bHighlight=1 and 3D objects with bHighlight=2 as mentioned <a href="#bHighlight">above</a>.<br>
<br>
<a name="createSurfaceRepresentation"></a><b>createSurfaceRepresentation(atoms, type, wireframe, opacity)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Create surface for "atoms". "type" can be 1 (Van der Waals surface), 2 (molecular surface), and 3 (solvent accessible surface). "wireframe" is a boolean to determine whether to show the surface as a mesh. "opacity" is a value between 0 and 1. "1" means not transparent at all. "0" means 100% transparent.<br>
<br>
<a name="createTube"></a><b>createTube(atoms, atomName, radius, bHighlight)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Create tubes for "atoms" with certain "atomName". "radius" is the radius of the tubes. "bHighlight" is an option to draw the highlight for these atoms. The highlight could be outlines with bHighlight=1 and 3D objects with bHighlight=2 as mentioned <a href="#bHighlight">above</a>.<br>
<br>
<a name="draw"></a><b>draw()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Draw the 3D structure. It rebuilds scene, applies previous color, applies the transformation, and renders the image.
<br>
<a name="drawCartoonNucleicAcid"></a><b>drawCartoonNucleicAcid(atoms, div, thickness, bHighlight)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Create curves for nucleotide "atoms". "div" means how many pnts are used to smooth the curve. It's typically 5. "thickness" is the thickness of the curve. "bHighlight" is an option to draw the highlight for these atoms. The highlight could be outlines with bHighlight=1 and 3D objects with bHighlight=2 as mentioned <a href="#bHighlight">above</a>.<br>
<br>
<a name="drawNucleicAcidStick"></a><b>drawNucleicAcidStick(atoms, bHighlight)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Create sticks between two nucleotide curves for nucleotide "atoms". "bHighlight" is an option to draw the highlight for these atoms. The highlight could be outlines with bHighlight=1 and 3D objects with bHighlight=2 as mentioned <a href="#bHighlight">above</a>.<br>
<br>
<a name="drawSymmetryMates2"></a><b>drawSymmetryMates2()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Draw the biological unit assembly using the matrix.<br>
<br>
<a name="exportCanvas"></a><b>exportCanvas()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Export the current view as a PNG file.<br>
<br>
<a name="getAtomsWithinAtom"></a><b>getAtomsWithinAtom(allAtoms, targetAtoms, distance)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
This function returns atoms within a certain "distance" (in angstrom) from the "targetAtoms". The returned atoms are stored in a hash with atom indices  as keys and 1 as values. Only those atoms in "allAtoms" are considered.<br>
<br>
<a name="getChainsFromAtoms"></a><b>getChainsFromAtoms(atomHash)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Return the chain hash from the atom hash. The chain hash has the chainid as the key and 1 as the value.<br>
<br>
<a name="getExtent"></a><b>getExtent(atomlist)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
For a list of atoms, return an array containing three coordinates: minimum x- y- z- values, maximum x- y- z- values, and average x- y- z- values.<br>
<br>
<a name="getFirstAtomObj"></a><b>getFirstAtomObj(atomsHash)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Return the first atom in the atom hash, which has the atom serial number as the key.<br>
<br>
<a name="getLastAtomObj"></a><b>getLastAtomObj(atomsHash)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Return the last atom in the atom hash, which has the atom serial number as the key.<br>
<br>
<a name="getResiduesFromAtoms"></a><b>getResiduesFromAtoms(atomHash)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Return the residue hash from the atom hash. The residue hash has the resid as the key and 1 as the value.<br>
<br>
<a name="hasCovalentBond"></a><b>hasCovalentBond(atom1, atom2)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Determine whether atom1 and atom2 have covalent bond.<br>
<br>
<a name="hideHbonds"></a><b>hideHbonds()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Remove hydrogen bonds.<br>
<br>
<a name="init"></a><b>init()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
When users first load a structure, call this function to empty previous settings.<br>
<br>
<a name="loadPDB"></a><b>loadPDB(pdbData)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
This PDB parser feeds the viewer with the content of a PDB file, pdbData.<br>
<br>
<a name="rebuildScene"></a><b>rebuildScene(options)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
This core function sets up the scene and display the structure according to the input options (shown <a href="#DisplayOptions">above</a>), which is a hash containing values for different keys. <br>

<br>
<a name="reinitAfterLoad"></a><b>reinitAfterLoad()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Reset parameters for displaying the loaded structure.<br>
<br>
<a name="removeHlObjects"></a><b>removeHlObjects()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Remove the highlight. The atom selection does not change.<br>
<br>
<a name="removeSurfaces"></a><b>removeSurfaces()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Remove previously drawn surfaces.<br>
<br>
<a name="render"></a><b>render()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Render the scene and objects into pixels.<br>
<br>
<a name="resetOrientation"></a><b>resetOrientation()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Set the orientation to the original one, but leave the style, color, etc alone.<br>
<br>

<a name="residueName2Abbr"></a><b>residueName2Abbr(residueName)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Convert a three-letter residue name to a one-letter residue abbreviation, e.g., 'LYS' to 'K', or '  A' to 'A' for nucleotides.<br>
<br>

<a name="rotateDown"></a><b>rotateDown(degree)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Rotate the structure certain degree to the bottom, e.g., 5 degree. <br>
<br>
<a name="rotateLeft"></a><b>rotateLeft(degree)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Rotate the structure certain degree to the left, e.g., 5 degree. <br>
<br>
<a name="rotateRight"></a><b>rotateRight(degree)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Rotate the structure certain degree to the right, e.g., 5 degree. <br>
<br>

<a name="rotateUp"></a><b>rotateUp(degree)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Rotate the structure certain degree to the top, e.g., 5 degree. <br>
<br>

<a name="saveFile"></a><b>saveFile(filename, type)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Save the state file or the image file with "filename". "type" is either "text" for state file or "png" for image file.<br>
<br>

<a name="selectStrandHelixFromAtom"></a><b>selectStrandHelixFromAtom(atom)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
For an "atom", select all atoms in the same strand, helix, or coil.<br>
<br>

<a name="setAtomStyleByOptions"></a><b>setAtomStyleByOptions(options)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Set atom style according to the definition in options (options.secondaryStructure, etc).<br>
<br>

<a name="setCamera"></a><b>setCamera()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Set the camera according to the size of the structure.<br>
<br>

<a name="setColorByOptions"></a><b>setColorByOptions(options, atoms)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Set atom color according to the definition in options (options.color).<br>
<br>

<a name="setOutlineColor"></a><b>setOutlineColor(colorStr)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Set the outline color when highlighting atoms. The available options are "yellow", "green", and "red". <br>
<br>

<a name="setRotationCenter"></a><b>setRotationCenter(coord)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Set the center at the position with coordinated "coord". <br>
<br>

<a name="setStyle2Atoms"></a><b>setStyle2Atoms(atomlist)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
For a list of atoms, set the hash with style as key and atom serial as value. <br>
<br>

<a name="setWidthHeight"></a><b>setWidthHeight(width, height)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Set the width and height of the canvas. <br>
<br>



<a name="showPicking"></a><b>showPicking(atom)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Define actions when an atom is picked. By default, the atom information ($[structure id].[chain id]:[residue number]@[atom name]) is displayed.<br>
<br>

<a name="translateDown"></a><b>translateDown(percentScreenSize)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Translate the structure certain distance to the bottom, e.g., "percentScreenSize" 1 means 1% of the screen height. <br>
<br>
<a name="translateLeft"></a><b>translateLeft(percentScreenSize)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Translate the structure certain distance to the left, e.g., "percentScreenSize" 1 means 1% of the screen width. <br>
<br>
<a name="translateRight"></a><b>translateRight(percentScreenSize)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Translate the structure certain distance to the right, e.g., "percentScreenSize" 1 means 1% of the screen width. <br>
<br>
<a name="translateUp"></a><b>translateUp(percentScreenSize)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Translate the structure certain distance to the top, e.g., "percentScreenSize" 1 means 1% of the screen height. <br>
<br>

<a name="zoomIn"></a><b>zoomIn(normalizedFactor)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Zoom in the structure at certain ratio, e.g., 0.1 is a reasonable value. <br>
<br>
<a name="zoominSelection"></a><b>zoominSelection()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Center on the selected atoms and zoom in.<br>
<br>
<a name="zoomOut"></a><b>zoomOut(normalizedFactor)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Zoom out the structure at certain ratio, e.g., 0.1 is a reasonable value. <br>
<br>

<span style="font-weight:bold; font-size:1em;">Hash Operations</span> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
<br>
<a name="cloneHash"></a><b>cloneHash(fromHash)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Clone the "fromHash" and return the cloned hash.<br>
<br>
<a name="exclHash"></a><b>exclHash(includeAtoms, excludeAtoms)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Get atoms in "includeAtoms", but not in "excludeAtoms". The returned hash has atom index as key and 1 as value.<br>
<br>
<a name="exclHash2Atoms"></a><b>exclHash2Atoms(includeAtoms, excludeAtoms)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Get atoms in "includeAtoms", but not in "excludeAtoms". The returned hash has atom index as key and atom object as value.<br>
<br>
<a name="hash2Atoms"></a><b>hash2Atoms(hash)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
The input "hash" has atom index as key and 1 as value. The returned hash has atom index as key and atom object as value.<br>
<br>
<a name="intHash"></a><b>intHash(atoms1, atoms2)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Get the intersection of two hashes "atoms1" and "atoms2". The returned hash has atom index as key and 1 as value.<br>
<br>
<a name="intHash2Atoms"></a><b>intHash2Atoms(atoms1, atoms2)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Get the intersection of two hashes "atoms1" and "atoms2". The returned hash has atom index as key and atom object as value.<br>
<br>
<a name="unionHash"></a><b>unionHash(atoms1, atoms2)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Get the union of two hashes "atoms1" and "atoms2". The returned hash has atom index as key and 1 as value.<br>
<br>
<a name="unionHash2Atoms"></a><b>unionHash2Atoms(atoms1, atoms2)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Get the union of two hashes "atoms1" and "atoms2". The returned hash has atom index as key and atom object as value.<br>
<br>

    <a name="UIMethods"></a>
    <h2>Method Documents of the advanced UI library iCn3DUI<img src="https://www.ncbi.nlm.nih.gov/Structure/IMG/spacer.gif" width="25" height="1" border="0"><a href="#Top"><img src="https://www.ncbi.nlm.nih.gov/Structure/IMG/arrowup_blue.gif" width="12" height="12" border="0" alt="back to top"></a></h2>

<h3>Options <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a></h3>
<br>
<a name="b2DShown"></a><b>b2DShown</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
A flag to remember whether the 2D interaction diagram was set.<br>
<br>
<a name="bAddCommands"></a><b>bAddCommands</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
A flag to determine whether to add current step into the command history.<br>
<br>
<a name="bAddLogs"></a><b>bAddLogs</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
A flag to determine whether to add current step into the log window.<br>
<br>
<a name="bAnnoShown"></a><b>bAnnoShown</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
A flag to remember whether the annotation window was set.<br>
<br>
<a name="bSetChainsAdvancedMenu"></a><b>bSetChainsAdvancedMenu</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
A flag to remember whether the menu of defined sets was set.<br>
<br>
<a name="bNotLoadStructure"></a><b>bNotLoadStructure</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
A flag to determine whether to load the coordinates of the structure. When resetting the view, it is true so that the coordinates of the structure will not be loaded again.<br>
<br>
<a name="cfg"></a><b>cfg</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
A hash containing all input parameters.<br>
<br>
<a name="LESSHEIGHT"></a><b>LESSHEIGHT</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
The height (in px) that was left empty by the 3D viewer. The default is 20px.<br>
<br>
<a name="LESSWIDTH"></a><b>LESSWIDTH</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
The width (in px) that was left empty by the 3D viewer. The default is 20px.<br>
<br>
<a name="options"></a><b>opts</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
The default display options as shown <a href="#DisplayOptions">above</a>.<br>
<br>
<a name="pre"></a><b>pre</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
A prefix for all custom html element id. It ensures all html elements have specific ids, even when multiple iCn3D viewers are shown together.<br>
<br>

<h3>Methods <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a></h3>
<br>

<a name="addChainLabels"></a><b>addChainLabels(atomHash)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Display chain name in the 3D structure display for the chains intersecting with the atoms in "atomHash".

<br><br><a name="addCustomSelection"></a><b>addCustomSelection(residueAtomArray, commandname, commanddesc, command, bResidue)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Define a custom selection based on the array of residues or atoms. The custom selection is defined by the "command" with the name "commandname" and the description "commanddesc". If "bResidue" is true, the custom selection is based on residues. Otherwise, the custom selection is based on atoms.

<br><br><a name="addLabel"></a><b>addLabel(text, x, y, z, size, color, background)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Add a "text" at the position (x, y, z) with the input "size", "color", and "background".

<br><br><a name="addLine"></a><b>addLine(x1, y1, z1, x2, y2, z2, color, dashed)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Add a line between the position (x1, y1, z1) and the position (x2, y2, z2) with the input "color". The line can be dashed if "dashed" is set true.

<br><br><a name="addStabilizer"></a><b>addStabilizer()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Add stabilizers in the model for 3D printing. This is especially important for the cartoon display such as ribbons.

<a name="addTerminiLabels"></a><b>addTerminiLabels(atomHash)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Display the terminal labels for the atoms in "atomHash". The termini of proteins are labeld as "N-" and "C-". The termini of nucleotides are labeled as "5'" and "3'".

<br><br><a name="allCustomEvents"></a><b>allCustomEvents()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
You can add your custom events in this function if you want to add new links in the function setTools.

<br><br><a name="allEventFunctions"></a><b>allEventFunctions()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Hold all functions related to click events.

<br><br><a name="alternateStructures"></a><b>alternateStructures()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Show structures one by one.

<br><br><a name="applyCommand"></a><b>applyCommand(command)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Execute a command. If the command is to load a structure, use the Method "applyCommandLoad".

<br><br><a name="applyCommandAnnotationsAndCddSite"></a><b>applyCommandAnnotationsAndCddSite()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
The annotation window calls many Ajax calls. Thus the command "view interactions" (in Share Link or loading state file) is handled specially to wait for the Ajax calls to finish before executing the next command.

<br><br><a name="applyCommandMap"></a><b>applyCommandMap(command)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Apply the command to show electron density map.

<br><br><a name="applyCommandEmmap"></a><b>applyCommandEmmap(command)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Apply the command to show EM density map.

<br><br><a name="applySavedColor"></a><b>applySavedColor()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Restore the previously saved color.

<br><br><a name="applySavedStyle"></a><b>applySavedStyle()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Restore the previously saved style.

<br><br><a name="applyCommandLoad"></a><b>applyCommandLoad(command)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Execute the command to load a structure. This step is different from the rest steps since it has to finish before the rest steps start.

<br><br><a name="back"></a><b>back()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Go back one step. Basically the commands are sequentially executed, but with one less step.


<br><br><a name="changeCustomAtoms"></a><b>changeCustomAtoms(nameArray)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
HighlightAtoms are set up based on the selected custom names "nameArray" in the atom menu. The corresponding atoms are neither highlighted in the sequence dialog nor in the 3D structure since not all residue atom are selected.

<br><br><a name="changeSeqColor"></a><b>changeSeqColor(residueArray)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Change the residue color in the annotation window for the residues in the array "residueArray".

<br><br><a name="clearSelection"></a><b>clearSelection()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Clear the "Structure", "Chain", and "Custom" dropdown menu.

<br><br><a name="deleteSelectedSets"></a><b>deleteSelectedSets()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Delete selected sets in the menu of "Defined Sets".

<br><br><a name="downloadAlignment"></a><b>downloadAlignment(align)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Load the VAST+ structure alignment for the pair of structures "align", e.g., "align" could be "1HHO,4N7N".

<br><br><a name="downloadBlast_rep_id"></a><b>downloadBlast_rep_id(sequence_id_comma_structure_id)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Ajax call was used to get the atom data from "sequence_id_comma_structure_id", comma-separated NCBI protein accessions of a protein sequence and a chain of a 3D structure (e.g., 23491729,1TUP_A). This function was deferred so that it can be chained together with other deferred functions for sequential execution. Note that only one structure corresponding to the blast_rep_id will be shown. If there is no structures available for the blast_rep_id, a warning message will be shown.

<br><br><a name="downloadCid"></a><b>downloadCid(cid)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Ajax call was used to get the atom data from the PubChem "cid". This function was deferred so that it can be chained together with other deferred functions for sequential execution.

<br><br><a name="downloadGi"></a><b>downloadGi(gi)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Ajax call was used to get the atom data from the NCBI "gi". This function was deferred so that it can be chained together with other deferred functions for sequential execution. Note that only one structure corresponding to the gi will be shown. If there is no structures available for the gi, a warning message will be shown.

<br><br><a name="downloadMmcif"></a><b>downloadMmcif(mmcifid)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Ajax call was used to get the atom data from the "mmcifid". This function was deferred so that it can be chained together with other deferred functions for sequential execution.


<br><br><a name="downloadMmdb"></a><b>downloadMmdb(mmdbid)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Ajax call was used to get the atom data from the NCBI "mmdbid". This function was deferred so that it can be chained together with other deferred functions for sequential execution. If the structure is too large, a 3D dgm will show up. You can select your interested chains to see the details.<br>

<br>Atom "data" from MMDB file was parsed to set up <a href="#DataInputs">parameters for the 3D viewer</a> by calling the function loadAtomDataIn. The deferred parameter was resolved after the parsing so that other javascript code can be executed.

<br><br><a name="downloadMmtf"></a><b>downloadMmtf(mmtfid)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
MMTF is a new binary data format besides the previous PDB and mmCIF formats for 3D structures. The JavaScript Methods at <a href="http://mmtf.rcsb.org/">http://mmtf.rcsb.org/</a> was used to load and parse the data.

<br><br><a name="downloadPdb"></a><b>downloadPdb(pdbid)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Ajax call was used to get the atom data from the "pdbid". This function was deferred so that it can be chained together with other deferred functions for sequential execution. A wrapper was added to support both http and https.

<br><br><a name="downloadUrl"></a><b>downloadUrl(URL, type)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Load structures from a "URL". Due to the same domain policy of Ajax call, the URL should be in the same domain. "type" could be "pdb", "mol2", "sdf", or "xyz" for pdb file, mol2file, sdf file, and xyz file, respectively.

<br><br><a name="execCommands"></a><b>execCommands(steps)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Execute a list of commands. "steps" is the total number of commands.

<br><br><a name="exportInteractions"></a><b>exportInteractions()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Export the list of residues in some chain interacting with residues in another chain.

<br><br><a name="forward"></a><b>forward()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Go forward one step. Basically the commands are sequentially executed, but with one more step.

<br><br><a name="getAlignSequencesAnnotations"></a><b>getAlignSequencesAnnotations(alignChainArray, bUpdateHighlightAtoms, residueArray, bShowHighlight)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Set up the sequence display with the aligned sequences. Either chains in "alignChainArray" or residues in "residueArray" will be highlighted. "bUpdateHighlightAtoms" is a flag to update the highlight atoms or not. "bShowHighlight" is a flag to show highlight or not.

<br><br><a name="getCommandsBeforeCrash"></a><b>getCommandsBeforeCrash()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Set the commands before the browser crashed. These commands are used to restore your previous state by refreshing the crashed page. It works in Chrome, Firefox, and Internet Explorer in PC, but neither Safari nor Mac.

<br><br><a name="hideMenu"></a><b>hideMenu(width, height)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Hide the menu at the top and just show the canvas. "width" and "height" are the width and height of the canvas.

<br><br><a name="hideStabilizer"></a><b>hideStabilizer()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Remove all the added stabilizers.

<br><br><a name="initUI"></a><b>init()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
When users load a state file, call this function to close all dialogs and reset the flags of these dialogs.<br>

<br><br><a name="hideStabilizer"></a><b>hideStabilizer()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Remove stabilizers in the model.

<br><br><a name="isIE"></a><b>isIE()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Determine whether the current browser is Internet Explorer.

<br><br><a name="isMac"></a><b>isMac()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Determine whether it is a Mac.

<br><br><a name="isMobile"></a><b>isMobile()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Determine whether it is a mobile device.

<br><br><a name="isSessionStorageSupported"></a><b>isSessionStorageSupported()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Determine whether Session Storage is supported in your browser. Session Storage is not supported in Safari.

<br><br><a name="loadAtomDataIn"></a><b>loadAtomDataIn(data, id, type)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
This function was used to parse atom "data" to set up <a href="#DataInputs">parameters for the 3D viewer</a>. "type" is mmcifid or mmdbid. "id" is the MMDB ID or mmCIF ID.

<br><br><a name="loadSdfAtomData"></a><b>loadSdfAtomData(data)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Atom "data" from SDF file was parsed to set up <a href="#DataInputs">parameters for the 3D viewer</a>. The deferred parameter was resolved after the parsing so that other javascript code can be executed.

<br><br><a name="loadMmcifData"></a><b>loadMmcifData(data)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Atom "data" from mmCIF file was parsed to set up <a href="#DataInputs">parameters for the 3D viewer</a> by calling the function loadAtomDataIn. The deferred parameter was resolved after the parsing so that other javascript code can be executed.

<br><br><a name="loadPdbData"></a><b>loadPdbData(data)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Atom "data" from PDB file was parsed to set up <a href="#DataInputs">parameters for the 3D viewer</a>. The deferred parameter was resolved after the parsing so that other javascript code can be executed.

<br><br><a name="loadScript"></a><b>loadScript(commands)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Run commands one after another. The commands can be semicolon ';' or new line '\n' separated.

<br><br><a name="modifyIcn3d"></a><b>modifyIcn3d()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Hold all modification to the core iCn3D library.

<br><br><a name="modifyIcn3dshowPicking"></a><b>modifyIcn3dshowPicking()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Modify the function "showPicking" in the core iCn3D library.

<br><br><a name="openDialog"></a><b>openDialog(id, title)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Open a dialog to input parameters. "id" is the id of the div section holding the html content. "title" is the title of the dialog. The dialog can be out of the viewing area.

<br><br><a name="outputSelection"></a><b>outputSelection()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Output the selected residues in the residue dialog.

<br><br><a name="pickCustomSphere"></a><b>pickCustomSphere(distance)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Select a sphere around the highlight atoms with a predefined distance.

<br><br><a name="prepareFor3Dprint"></a><b>prepareFor3Dprint()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Prepare for 3D printing by changing dashed lines to solid lines, changing the thickness of the model.

<br><br><a name="removeHl2D"></a><b>removeHl2D()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Remove the highlight in the 2D interaction diagram.

<br><br><a name="removeHlAll"></a><b>removeHlAll()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Remove the highlight in 3D structure, 2D interaction, 1D sequence, and the menu of defined sets.

<br><br><a name="removeHlMenus"></a><b>removeHlMenus()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Remove the selection in the menu of defined sets.

<br><br><a name="removeHlObjects"></a><b>removeHlObjects()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Remove the highlight in the 3D structure display.

<br><br><a name="removeHlSeq"></a><b>removeHlSeq()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Remove the highlight in the sequence display of the annotation window.

<br><br><a name="removeSeqChainBkgd"></a><b>removeSeqChainBkgd()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Remove the background of the highlighted chain in the sequence dialog.

<br><br><a name="removeSeqResidueBkgd"></a><b>removeSeqResidueBkgd()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Remove the background of the highlighted residues in the sequence dialog.

<br><br><a name="renderFinalStep"></a><b>renderFinalStep(steps)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
When reading a list of commands, apply transformation at the last step.

<br><br><a name="renderStructure"></a><b>renderStructure()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Update the dropdown menu and show the structure by calling the function "draw()".

<br><br><a name="resetAfter3Dprint"></a><b>resetAfter3Dprint()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Reset the hydrogen bonds, distance lines to dashed lines. Reset the thickness to the default values.

<br><br><a name="resizeCanvas"></a><b>resizeCanvas(width, height)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Resize the canvas with the defined "width" and "height".

<br><br><a name="rotStruc"></a><b>rotStruc(direction)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Rotate the structure in one of the directions: "left", "right", "up", and "down".


<br><br><a name="saveCommandsToSession"></a><b>saveCommandsToSession()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Save the command history to session storage so that the viewer can show the previous state when refreshing the same page.

<br><br><a name="saveColor"></a><b>saveColor()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Save the current color setting so that these colors can be restored later by clicking "Apply Saved Color" in the Color menu.

<br><br><a name="saveFile"></a><b>saveFile(filename, type, data)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Five types are used: command, png, html, text, and binary. The type "command" is used to save the statefile. The type "png" is used to save the current canvas image. The type "html" is used to save html file with the "data". This can be used to save any text. The type "text" is used to save an array of text, where "data" is actually an array. The type "binary" is used to save an array of binary, where "data" is actually an array.

<br><br><a name="saveStyle"></a><b>saveStyle()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Save the current style setting so that these styles can be restored later by clicking "Apply Saved Style" in the Style menu.

<br><br><a name="saveStlFile"></a><b>saveStlFile()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Save the binary STL file for 3D monocolor printing.

<br><br><a name="saveVrmlFile"></a><b>saveVrmlFile()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Save the VRML file for 3D color printing.

<br><br><a name="selectAAlignChain"></a><b>selectAAlignChain(alignChainid, commandname)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Select an aligned chain with the chain id "alignChainid" in the sequence dialog and save it as a custom selection with the name "commandname".

<br><br><a name="selectAChain"></a><b>selectAChain(chainid, commandname)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Select a chain with the chain id "chainid" in the sequence dialog and save it as a custom selection with the name "commandname".

<br><br><a name="selectAll"></a><b>selectAll()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Select all atom in the structures.

<br><br><a name="selectByCommand"></a><b>selectByCommand(command, commandname, commanddesc)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Set a custom selection with the "command", its name "commandname" and its description "commanddesc".

<br><br><a name="selectComplement"></a><b>selectComplement()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Select the complement of the current selection.

<br><br><a name="selectSideChains"></a><b>selectSideChains()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Select only the side chain atoms of the current selection.

<br><br><a name="set2DDiagrams"></a><b>set2DDiagrams(mmdbid)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Generate the 2D interaction diagram for the structure "mmdbid", which could be PDB ID. The 2D interaction diagram is only available when the input is NCBI MMDB ID, i.e., the URL is something like "&mmdbid=...".

<br><br><a name="setAlignChainMenu"></a><b>setAlignChainMenu(bInitial)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Set the aligned chain menu. "bInitial" is a flag for the initial loading.

<br><br><a name="setAtomMenu"></a><b>setAtomMenu(commandnameArray)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Set the menu of defined sets with an array of defined names "commandnameArray".

<br><br><a name="setCustomDialogs"></a><b>setCustomDialogs()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
A placeholder for all custom dialogs.

<br><br><a name="setDialogs"></a><b>setDialogs()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Set the html for all popup dialogs.

<br><br><a name="setLogCmd"></a><b>setLogCmd(command, bSetCommand)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Show the input command in log. If "bSetCommand" is true, the command will be saved in the state file as well.

<br><br><a name="setLogWindow"></a><b>setLogWindow()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Set the textarea for the log output.

<br><br><a name="setMenu1"></a><b>setMenu1()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Set the menu "File" at the top of the viewer.

<br><br><a name="setMenu2"></a><b>setMenu2()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Set the menu "Select" at the top of the viewer.

<br><br><a name="setMenu3"></a><b>setMenu3()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Set the menu "Style" at the top of the viewer.

<br><br><a name="setMenu4"></a><b>setMen4()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Set the menu "Color" at the top of the viewer.

<br><br><a name="setMenu5"></a><b>setMenu5()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Set the menu "Surface" at the top of the viewer.

<br><br><a name="setMenu6"></a><b>setMenu6()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Set the menu "Other" at the top of the viewer.


<br><br><a name="setOption"></a><b>setOption(id, value)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Modify the <a href="#DisplayOptions">display options</a>, e.g., setOption('color', 'green')

<br><br><a name="setStyle"></a><b>setStyle(selectionType, style)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Set the styles of predefined "protein", "nucleotides", etc.

<br><br><a name="setTools"></a><b>setTools()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Set the HTML code for the tools section. It includes several buttons, and is the second line at the top of the viewer.

<br><br><a name="setTopMenusHtml"></a><b>setTopMenus()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Set the HTML code for the menus shown at the top of the viewer.

<br><br><a name="shareLink"></a><b>shareLink()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Generate a URL to capture the current state and open it in a new window. Basically the state file (the comand history) is concatenated in the URL to show the current state.


<br><br><a name="show3DStructure"></a><b>show3DStructure()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
The main function to show the 3D structure.

<br><br><a name="showAnnotations"></a><b>showAnnotations()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
show annotations such as SNPs, ClinVar, domains, binding sites, etc.

<br><br><a name="showCddSite"></a><b>showCddSite(chainid, chainidBase)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Show the annotations of CDD domains and binding sites.

<br><br><a name="showDomain"></a><b>showDomain(chainid, chainidBase)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Show the annotations of 3D domains.

<br><br><a name="showHbonds"></a><b>showHbonds(threshold)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Show the hydrogen bonds between chemicals and proteins/nucleotides with dashed-lines."threshold" defines the distance of hydrogen bonds.

<br><br><a name="showInteraction"></a><b>showInteraction(chainid, chainidBase)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Show the residues interacting with the chain.

<br><br><a name="showMenu"></a><b>showMenu(width, height)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Show the menu at the top and the canvas. "width" and "height" are the width and height of the canvas.

<br><br><a name="showSelection"></a><b>showSelection()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Show the selection.

<br><br><a name="showSeq"></a><b>showSeq(chainid, chainidBase)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Show the sequences and secondary structures.

<br><br><a name="showSets"></a><b>showSets()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Display the menu of defined sets. All chains and defined custom sets are listed in the menu. All new custom sets will be displayed in the menu.

<br><br><a name="showSnpClinvar"></a><b>showSnpClinvar(chainid, chainidBase)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Show the annotations of SNPs and ClinVar.

<br><br><a name="showSsbonds"></a><b>showSsbonds()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Show the disulfide bonds and show the side chain in the style of "stick".

<br><br><a name="showTitle"></a><b>showTitle()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Show the title and PDB ID of the PDB structure at the beginning of the viewer.

<br><br><a name="switchHighlightLevelDown"></a><b>switchHighlightLevelDown()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
When users pick an atom, a residue, a strand/helix, a chain, or a structure, they can use upper arrow to increase the highlight level by one, or use down arrow to decrease the highlight level by one. This function switchHighlightLevelDown() decreases the highlight level by one.

<br><br><a name="switchHighlightLevelUp"></a><b>switchHighlightLevelUp()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
When users pick an atom, a residue, a strand/helix, a chain, or a structure, they can use upper arrow to increase the highlight level by one, or use down arrow to decrease the highlight level by one. This function switchHighlightLevelUp() increases the highlight level by one.

<br><br><a name="toggleSelection"></a><b>toggleSelection()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Toggle on and off the current selection.

<br><br><a name="update2DdgmContent"></a><b>update2DdgmContent()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
The 2D diagram only shows the currently displayed chains when users click the option "View Only Selection". This method is called to dynamically update the content of the 2D interaction diagram.

<br><br><a name="updateHl2D"></a><b>updateHl2D()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Update the highlight of 2D interaction diagram according to the current highlighted atoms.

<br><br><a name="updateHlAll"></a><b>updateHlAll(commandnameArray, bSetMenu)</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Update the highlight of 3D structure, 2D interaction, sequences, and the menu of defined sets according to the current highlighted atoms.

<br><br><a name="updateHlMenus"></a><b>updateHlMenus()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Update the selection in the menu of defined sets according to the current highlighted atoms.

<br><br><a name="updateHlObjects"></a><b>updateHlObjects()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Update the highlight of 3D structure display according to the current highlighted atoms.

<br><br><a name="updateHlSeq"></a><b>updateHlSeq()</b> <a href="#Top"><span class="top">&nbsp;&nbsp;&nbsp;</span></a><br>
Update the highlight of sequences in the annotation window according to the current highlighted atoms.

<br><br>

<hr>

<a name="log"></a>
<h2>Change Log:<img src="https://www.ncbi.nlm.nih.gov/Structure/IMG/spacer.gif" width="25" height="1" border="0"><a href="#Top"><img src="https://www.ncbi.nlm.nih.gov/Structure/IMG/arrowup_blue.gif" width="12" height="12" border="0" alt="back to top"></a></h2>
<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.17.4.zip">icn3d-2.17.4</a> was release on June 24, 2020. Users now can select options for cross structure interaction in alignment. Get ClinVar annotations from database.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.17.3.zip">icn3d-2.17.3</a> was release on June 18, 2020. Added the track "Cross-Linkages" for glycans, etc.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.17.2.zip">icn3d-2.17.2</a> was release on June 17, 2020. Added pinger to log usage.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.17.1.zip">icn3d-2.17.1</a> was release on June 15, 2020. Enabled to accept RID from BLAST result page.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.17.0.zip">icn3d-2.17.0</a> was release on June 12, 2020. Users can now show "2D Interaction Graph" and also "Highlight Interactions in Table" in the menu "View > H-Bonds & Interactions". Users can also replay the share link step by step to learn how to generate a custom display in the menu "File > Replay Each Step".
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.16.4.zip">icn3d-2.16.4</a> was release on June 8, 2020. Enabled to turn force off in 2D Graph so that users can manually arrange the nodes.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.16.3.zip">icn3d-2.16.3</a> was release on June 5, 2020. Added halogen bonds, pi-cation, and pi-stacking interactions in the menu "View > H-Bonds & Interaction" and showed the interactions in sorted tables.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.16.2.zip">icn3d-2.16.2</a> was release on May 29, 2020. Added the following features: 1. save contents in any dialog/window using the save icon next to the closeicon, 2. set names for each tab/window using the menu "Windows > Your Note / Window Title", 3. select by property (such as residue type, solvent accessibilty, etc) in the "Select" menu, 4. Use a custom file to define the color or tube size for each residue by clicking the button "Custom Color / Tube" in the "Sequences & Annotations" window.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.16.1.zip">icn3d-2.16.1</a> was release on May 27, 2020. Enabled to show gaps when adding multiple sequence alignment data as tracks.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.16.0.zip">icn3d-2.16.0</a> was release on May 21, 2020. Enabled to show interactions using force-directed graph in the menu "View > H-Bonds & Interactions > Force-Directed Graph". Enabled to calculate Solvent Accessible Surface Area (SASA) in the menu "View > Surface Area", or color by SASA in the menu "Color > Solvent Accessibility". Fixed ClinVar and SNP annotations.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.15.3.zip">icn3d-2.15.3</a> was release on May 8, 2020. Enabled to update the short "Share Link" URL; enabled to save in a sharable URL the "Side by Side" view, which is useful to view two aligned structures.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.15.2.zip">icn3d-2.15.2</a> was release on May 5, 2020. Fixed the display of electron density map and EM map.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.15.1.zip">icn3d-2.15.1</a> was release on April 27, 2020. Improved the unionHash function to speed up selection. Improved the message when loading a list of commands.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.15.0.zip">icn3d-2.15.0</a> was release on April 21, 2020. Enabled to show the same structure "Side by Side" in two views in the "View" menu. Each view has the same orientation, but can have independent 3D display. Enabled to add multiple sequence alignments as tracks when clicking "Add Track" in the "Sequences & Annotations" window. Added "Hide Selection" in the "View" menu. Improved selection on "H-Bonds & Interactions". Improved the UI for "Realign Selection" in the "File" menu. The gallery shows COVID-19-related structures at the top.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.14.0.zip">icn3d-2.14.0</a> was release on April 15, 2020. Added the features to load Electron Density data in the menu "File > Open File > Electron Density (DSN6)", resize the 3D window, realign two structures in the menu "File > Realign", color residues with custom colors in the menu "Color > Residue > Custom", and add custom colors when aligning a sequence to a structure.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.13.1.zip">icn3d-2.13.1</a> was release on March 26, 2020. Showed membranes for transmembrane proteins in VAST+ alignment.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.13.0.zip">icn3d-2.13.0</a> was release on March 23, 2020. Added the "Symmetry" feature in the View menu.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.12.1.zip">icn3d-2.12.1</a> was release on March 12, 2020. Fixed a bug on hydrogen bonds between residues with the same residue number but different chain names. Removed duplicated names in the menu of "Defined Sets".
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.12.0.zip">icn3d-2.12.0</a> was release on March 2, 2020. Enabled to realign selected residues in VAST+ structure alignment. The option "Realign Selection" is in the View menu.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.11.5.zip">icn3d-2.11.5</a> was release on February 11, 2020. Add colors to aligned sequence track based on Blosum62.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.11.4.zip">icn3d-2.11.4</a> was release on January 14, 2020. The compiling tool gulp was upgraded to version 4.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.11.3.zip">icn3d-2.11.3</a> was release on January 7, 2020. A bug was fixed to have predefined sets available for hydrogen bonds/interations.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.11.2.zip">icn3d-2.11.2</a> was release on December 20, 2019. Added the style Backbone and added 'use strict' to each function.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.11.1.zip">icn3d-2.11.1</a> was release on December 19, 2019. Fixed a bug in chain alignment due to the introduction of membranes.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.11.0.zip">icn3d-2.11.0</a> was release on December 16, 2019. Enabled to show membranes for transmembrane proteins for data from PDB or MMDB by aligning the coordinates to the data from Orientations of Proteins in Membranes (OPM) database. Users could adjust the location of the membrane as well.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.10.1.zip">icn3d-2.10.1</a> was release on December 9, 2019. Enabled to select the currently displayed set in the Select menu and fixed some bugs on H-Bonds & Interactions.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.10.0.zip">icn3d-2.10.0</a> was release on December 5, 2019. Enabled to show each hydrogen bond and contact in 3D using the menu "View > H-Bonds & Interactions".
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.9.2.zip">icn3d-2.9.2</a> was release on November 18, 2019. Added Transmembrane track if the input is opmid. Added angle constraint for hydrogen bonds.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.9.1.zip">icn3d-2.9.1</a> was release on November 14, 2019. Enable to select regions between two X-Y membranes for transmembrane proteins.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.9.0.zip">icn3d-2.9.0</a> was release on November 8, 2019. Display membranes for transmembrane proteins using data from Orientations of Proteins in Membranes (OPM). The feature is at "File > Retrieve by ID > OPM PDB ID".
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.8.3.zip">icn3d-2.8.3</a> was release on November 5, 2019. Display/output salt bridges; color helices and sheets with spectrum in the menu "Color > Secondary > Spectrum".
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.8.2.zip">icn3d-2.8.2</a> was release on October 29, 2019. Reduced the size of three.js (version 103) library. Added links to dbSNP in the mouseover texts of SNP annotations.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.8.1.zip">icn3d-2.8.1</a> was release on October 21, 2019. Fixed the 2D interaction display in structure alignment. The bug was introduced in the last release.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.8.0.zip">icn3d-2.8.0</a> was release on October 10, 2019. Allowed to align any chain to another chain in the menu "File > Align > Chain to Chain".
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.7.19.zip">icn3d-2.7.19</a> was release on September 9, 2019. Fixed the input width and height with "%".
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.7.18.zip">icn3d-2.7.18</a> was release on September 4, 2019. Added the option to view in Full Screen mode by clicking the expansion icon in the top-right corner when "mobilemenu" is turned on, or by clicking "Full Screen" in the View menu.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.7.17.zip">icn3d-2.7.17</a> was release on September 3, 2019. Made it easy to save interactive work by clicking "File > Save Files > iCn3D PNG Images". This saves both "iCn3D PNG Image" and an HTML file with a clickable PNG image, which is link to the custom display via a sharable link.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.7.16.zip">icn3d-2.7.16</a> was release on August 29, 2019. Fixed the transparent display by switching three.js from version 107 to 103.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.7.15.zip">icn3d-2.7.15</a> was release on August 21, 2019. Added an example page to embed multiple iCn3D viewers in one page.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.7.14.zip">icn3d-2.7.14</a> was release on August 21, 2019. Enabled the mobile style menu with the URL parameter "mobilemenu=1".
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.7.13.zip">icn3d-2.7.13</a> was release on August 14, 2019. Fixed the calculations on contacting atoms by considering the centers of atoms and not radii of atoms. Minimized the code size by retrieving some rarely used code on the fly.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.7.12.zip">icn3d-2.7.12</a> was release on August 8, 2019. The backend to retrieve ClinVar annotation was fixed.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.7.11.zip">icn3d-2.7.11</a> was release on August 6, 2019. Added "Label Scale" in the View menu to scale all labels.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.7.10.zip">icn3d-2.7.10</a> was release on August 5, 2019. Improved the display of binding sites with fog and slab.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.7.9.zip">icn3d-2.7.9</a> was release on August 1, 2019. Fixed the effect of Fog on sticks and spheres.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.7.8.zip">icn3d-2.7.8</a> was release on July 31, 2019. Fixed SNP annotation in the sequences and annotations window.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.7.7.zip">icn3d-2.7.7</a> was release on July 30, 2019. Added the option to show or hide hydrogens when displaying PubChem compounds.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.7.6.zip">icn3d-2.7.6</a> was release on July 18, 2019. Enabled to show disulfide bonds when a custom pdb file is input; added the option to output pairs for disulfide bonds, hydrogen bonds, and interacting/contacting residues by distance.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.7.5.zip">icn3d-2.7.5</a> was release on July 16, 2019. Added a Gallery section. Fixed the picking and centering issues in some Mac computers. Optimized the width and height of embedded iCn3D viewer.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.7.4.zip">icn3d-2.7.4</a> was release on July 8, 2019. Included Miniland1333's fix on clickTab for the basic display; auto-detected lipids and treated them as chemicals; improved the display of modified PDB files; mouseover showed the structure names when there are more than one structures.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.7.3.zip">icn3d-2.7.3</a> was release on June 6, 2019. Improved the display of transparent surfaces.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.7.2.zip">icn3d-2.7.2</a> was release on June 5, 2019. Fixed the display of transparent surfaces.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.7.1.zip">icn3d-2.7.1</a> was release on June 3, 2019. Fixed the sequence display when there are insertion codes or missing coordinates.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.7.0.zip">icn3d-2.7.0</a> was release on May 20, 2019. "Share Link" can be used to reproduce a custom display when the input is a known ID. "iCn3D PNG Image" can be saved and opened in the File menu to reproduce a custom display for all cases, even when the input is a PDB file or other files.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.6.6.zip">icn3d-2.6.6</a> was release on May 16, 2019. The sequence display was fixed when the input is a MMTF ID or a mmCIF ID.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.6.5.zip">icn3d-2.6.5</a> was release on May 14, 2019. The sequence display was fixed when the input is a pdb file.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.6.4.zip">icn3d-2.6.4</a> was release on April 29, 2019. jQuery was upgraded to version 3.4.0.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.6.3.zip">icn3d-2.6.3</a> was release on April 23, 2019. Showed SNP annotations for more 3D structures.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.6.2.zip">icn3d-2.6.2</a> was release on April 15, 2019. Enabled to show large structure such as HIV-1 capsid (3J3Q) in the basic version of iCn3D (https://www.ncbi.nlm.nih.gov/Structure/icn3d/?mmtfid=3j3q).
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.6.1.zip">icn3d-2.6.1</a> was release on April 1, 2019. Enabled to link from BLAST result page to iCn3D.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.6.0.zip">icn3d-2.6.0</a> was release on March 19, 2019. Users can now align any sequence to a hit structure by clicking "Align > Sequence to Structure" in the File menu. The default color scheme is color by sequence "Conservation" for sequence-structure or structure-structure alignments.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.5.3.zip">icn3d-2.5.3</a> was release on March 12, 2019. Added commands for up arraow and down arrow after picking a residue.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.5.2.zip">icn3d-2.5.2</a> was release on March 5, 2019. The style Lines was fixed by replacing THREE.Line with THREE.LineSegments.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.5.1.zip">icn3d-2.5.1</a> was release on February 14, 2019. Change log was moved to the file CHANGELOG.md. Share Link was changed from https://d55qc.app.goo.gl/### to https://structure.ncbi.nlm.nih.gov/icn3d/share.html?###. All previous share links still work. iCn3D library file was renamed from full_ui_all_#.#.#.min.js to icn3d_full_ui_#.#.#.min.js. Both files are available to make it backward compatible.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.5.0.zip">icn3d-2.5.0</a> was release on January 31, 2019. Updated Three.js from version 80 to version 99. Enabled the basic version (simple_ui_all.min.js) to hide the Tools menu and title. Fixed a bug in picking an atom for distance or labeling.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.4.3.zip">icn3d-2.4.3</a> was release on January 22, 2019. Non-standard proteins or nucleotides were still displayed as "Biopolymer" in 2D interactions and were displayed in the style of protein or nucleotide in 3D. The usage tracking was implemented in iCn3D.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.4.2.zip">icn3d-2.4.2</a> was release on January 16, 2019. Non-standard proteins or nucleotides were displayed as "Biopolymer" in 2D interactions and were displayed in the style of "Stick" in 3D. A new kind of annotation "Disulfie Bonds" was added to the "Sequences and Annotations" window.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.4.1.zip">icn3d-2.4.1</a> was release on January 7, 2019. Enabled users to show EM density map for any subset of an EM structure.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.4.0.zip">icn3d-2.4.0</a> was release on December 17, 2018. Enabled users to show electron density map for any subset of a crystal structure.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.3.4.zip">icn3d-2.3.4</a> was release on December 12, 2018. Enabled users to load a saved iCn3D PNG image into iCn3D to reproduce the display using the URL embedded in the image.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.3.3.zip">icn3d-2.3.3</a> was release on December 6, 2018. Made the list of interacting residues consistent in "File -> Save Files -> Interaction List" and in the "Sequences and Annotations" window.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.3.2.zip">icn3d-2.3.2</a> was release on October 30, 2018. Water molecules were enabled to be shown when the structure is not a biological assembly. Gene symbols were shown for each chain in the "Sequences and Annotations" window.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.3.1.zip">icn3d-2.3.1</a> was release on October 25, 2018. The color of the the first residue in a coil was fixed.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.3.0.zip">icn3d-2.3.0</a> was release on October 18, 2018. Added set operations (union,intersection, exclusion) in "Defined Sets"; added buttons "Helix Sets" and "Sheet Sets" in the "Sequences and Annotations" window to define helix sets and sheet sets in the window "Defined Sets"; added "Save Color" and "Apply Saved Color" in the menu "Color"; added "Save Style" and "Apply Saved Style" in the menu "Style"; added "Side Chains" in the menu "Select" to select side chains; added two options for color by "Secondary" structures: "Sheets in Green" and "Sheets in Yellow"; added color by "B-factor" that is normalized with "Original" values or "Percentile" values.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.2.5.zip">icn3d-2.2.5</a> was release on September 17, 2018. A bug in loading local PDB file was fixed.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.2.4.zip">icn3d-2.2.4</a> was release on September 6, 2018. The location of 2D interaction dialog was optimized.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.2.3.zip">icn3d-2.2.3</a> was release on August 30, 2018. Added an option to show N- and C-terminal labels.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.2.2.zip">icn3d-2.2.2</a> was release on August 9, 2018. Defined sets can be combined using "or", "and", and "not".
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.2.1.zip">icn3d-2.2.1</a> was release on August 3, 2018. Mouseover on the 3D structure shows the residue or atom name. Some Ajax calls are combined into one Ajax call.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.2.0.zip">icn3d-2.2.0</a> was release on July 30, 2018. The smoothing algorithm was switched from Catmull-Rom spline to cubic spline to make the curves more smooth. The thickness of ribbon was decreased to make the sides of the ribbons less apparent. The radio buttons in the menus was replaced by the check sign. A "Save Image" button was added in the "Toolbar".
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.1.8.zip">icn3d-2.1.8</a> was release on July 12, 2018. Checked the code with the strict mode.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.1.7.zip">icn3d-2.1.7</a> was release on June 28, 2018. Simplified the addition of custom text as a track in the Sequences and Annotations window.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.1.6.zip">icn3d-2.1.6</a> was release on June 21, 2018. A color picker was added to the color menu.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.1.5.zip">icn3d-2.1.5</a> was release on June 18, 2018. 3D printing are enabled for biological assemblies.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.1.4.zip">icn3d-2.1.4</a> was release on June 7, 2018. The retrieval of transformation matrix from mmCIF was fixed for Mac.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.1.3.zip">icn3d-2.1.3</a> was release on May 30, 2018.  "Sequences and Annotations" is now able to be highlighted even if some annotations didn't show up.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.1.2.zip">icn3d-2.1.2</a> was release on May 23, 2018. The surface display was improved by adding light reflection. Light was added to the display of instanced biological assemblies.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.1.1.zip">icn3d-2.1.1</a> was release on May 22, 2018. The option of color by "Spectrum" was added back.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.1.0.zip">icn3d-2.1.0</a> was release on May 21, 2018. The instancing method is used to display a biological assembly. It significantly improved the rendering speed by sending only the geometry of its assymmetruic unit to GPU and applying transformation matrices to display the assembly.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.0.3.zip">icn3d-2.0.3</a> was release on May 2, 2018. Removed the "Description" field when saving a set of atoms. This made "Share Link" URL shorter. Made the size of stabilizer thicker for 3D printing.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.0.2.zip">icn3d-2.0.2</a> was release on April 30, 2018. Reset WebGLRenderer when WebGL context is lost in Internet Explore 11.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.0.1.zip">icn3d-2.0.1</a> was release on April 23, 2018. The bug about extra 3D domains in the "Sequences & Annotations" window was fixed. The stabilizers for 3D printing were improved.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-2.0.0.zip">icn3d-2.0.0</a> was release on April 17, 2018. By clicking the menu "Windows: View Sequences & Annotations", users can view all kinds of annotations: ClinVar, SNPs, CDD domains, 3D domains, binding sites, interactions, and custom tracks. Users can click the menu "View: Chemical Binding" to show the chemical binding sites. Users can also export files for 3D printing at the menu "File: 3D Printing: VRML (Color, W/ Stabilizers)".
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-1.4.1.zip">icn3d-1.4.1</a> was release on November 3, 2017. The version of THREE.js in the zip file was fixed.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-1.4.0.zip">icn3d-1.4.0</a> was release on November 2, 2017. The rendering speed has been significantly improved by using the Imposter shaders from NGL Viewer. A bug in "Share Link" was fixed.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-1.3.10.zip">icn3d-1.3.10</a> was release on October 27, 2017. The "Save File" issue in Chrome 60 and after was fixed.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-1.3.9.zip">icn3d-1.3.9</a> was release on September 5, 2017. The handling of residues with insertion codes was fixed in structure alignment.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-1.3.8.zip">icn3d-1.3.8</a> was release on August 7, 2017. The handling of residues with insertion codes was fixed.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-1.3.7.zip">icn3d-1.3.7</a> was release on April 18, 2017. A bug in the output order of commands was fixed.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-1.3.6.zip">icn3d-1.3.6</a> was release on April 10, 2017. A bug introduced in the version of icn3d-1.3.5 was fixed in the function unionHash.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-1.3.5.zip">icn3d-1.3.5</a> was release on March 23, 2017. The codes were optimized to show 3D structures as soon as possible. Vast+ structure alignment was optimized as well.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-1.3.4.zip">icn3d-1.3.4</a> was release on March 1, 2017. The backend of structure alignment was updated.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-1.3.3.zip">icn3d-1.3.3</a> was release on November 15, 2016. Now users can save the image with "transparent" background using a single url, e.g., <a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/full.html?mmdbid=1tup&width=300&height=300&command=set%20background%20transparent;%20export%20canvas">https://www.ncbi.nlm.nih.gov/Structure/icn3d/full.html?mmdbid=1tup&width=300&height=300&command=set%20background%20transparent;%20export%20canvas</a>.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-1.3.2.zip">icn3d-1.3.2</a> was release on October 18, 2016. The atom specification in "Advanced set selection" was modified to use "$" instead of "#" in front of structure IDs. This modification avoids to the problem of showing multiple "#" in the urls of "Share Link".
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-1.3.1.zip">icn3d-1.3.1</a> was release on October 4, 2016. Partial diplay of helices or beta-sheets are enabled. The side chains, if displayed, are connected to C-alphas.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-1.2.3.zip">icn3d-1.2.3</a> was release on September 13, 2016. The MMTF format started to support https.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-1.2.2.zip">icn3d-1.2.2</a> was release on August 18, 2016. Added a switch button to switch between all atoms and selected atoms. When the mode is "selected atoms", the switch and the text "selection" next to it are colored in orange. The menu "Style", "Color", and "Surface" are colored in orange and only apply to selected atoms.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-1.2.1.zip">icn3d-1.2.1</a> was release on August 18, 2016. Some bugs were fixed.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-1.2.0.zip">icn3d-1.2.0</a> was release on August 17, 2016. The dialog of 2D interactions was added to show the interactions among different chains. Both the nodes (chains) and lines (interactions) can be selected. Secondary structures will be calculated if the input PDB file has no defined secondary structure information. The previous files src/icn3d.js, src/full_ui.js, and src/simple_ui.js were separated into small files.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-1.1.1.zip">icn3d-1.1.1</a> was release on July 25, 2016. Some bugs were fixed.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-1.1.0.zip">icn3d-1.1.0</a> was release on July 18, 2016. The new binary MMTF file format was supported. A new "Analysis" menu was added with an option to show disulfide bonds. Users can also input data from a url, either through the UI or through a encoded url parameter "?type=pdb&url=...", e.g., <a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/full.html?type=pdb&url=https%3A%2F%2Ffiles.rcsb.org%2Fview%2F1gpk.pdb">https://www.ncbi.nlm.nih.gov/Structure/icn3d/full.html?type=pdb&url=https%3A%2F%2Ffiles.rcsb.org%2Fview%2F1gpk.pdb</a>.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-1.0.1.zip">icn3d-1.0.1</a> was release on May 16, 2016.
<br><br>

<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-1.0.0.zip">icn3d-1.0.0</a> was release on April 28, 2016.
<br><br>

The beta version <a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-0.9.6-dev.zip">icn3d-0.9.6-dev</a> was release on April 21, 2016. Enabled to export and import selection file where each custom sets of atoms are defined. Javascript files and CSS files are versioned. Developers can use the default latest version or specify the specific version in their pages.
<br><br>

The beta version <a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-0.9.5-dev.zip">icn3d-0.9.5-dev</a> was release on April 4, 2016. Enabled to import Mol2, SDF, XYZ, PDB, and mmCIF files. Added "Schematic" style for chemicals. Improved the coordination between pk on 3D structure and selection on sequences.
<br><br>

The beta version <a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-0.9.4-dev.zip">icn3d-0.9.4-dev</a> was release on March 14, 2016. Added "Fog" and "Slab" features.
<br><br>

The beta version <a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-0.9.3-dev.zip">icn3d-0.9.3-dev</a> was release on March 9, 2016. Improved the following features: "Back" and "Forward" button, Export State, Open State.
<br><br>

The beta version <a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-0.9.2-dev.zip">icn3d-0.9.2-dev</a> was release on March 4, 2016. CSS namespace was added. The file simple_ui.js was reorganized to share some codes with full_ui.js. A "Schematic" style was added to show one letter residue name in the C-alpha (for protein) or O3' (for nucleotide) position.
<br><br>

The beta version <a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-0.9.1-dev.zip">icn3d-0.9.1-dev</a> was release on Feb 9, 2016. The surface generation was switched from the iview version (surface.js) to the more efficient 3Dmol version (ProteinSurface4.js and marchingcube.js).
<br><br>

The beta version <a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d-0.9.0-dev.zip">icn3d-0.9.0-dev</a> was release on Jan 17, 2016.

<br>

<!-- ======================= END_EXAMPLES ======================= -->

</blockquote>



  <!-- log & Google Analytics -->
  <script type="text/javascript" src="https://www.ncbi.nlm.nih.gov/core/pinger/pinger.js"></script>

</body></html>
